---
title: "AMF communities in DOK (SoilClim)"
author: "Katja Kozjek"
date: "`r Sys.Date()`"
output: 
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

# Prepare the environment

```{r setting the environment, message=F, warning=F, echo=F}

#clean everything
#rm(list=ls())

#load libraries
library(dada2); packageVersion("dada2") #1.14.0
library(phyloseq); packageVersion("phyloseq") #1.30.0
library(vegan); packageVersion("vegan") #2.5.6
library(ggplot2); packageVersion("ggplot2") #3.2.1
library(dplyr); packageVersion("dplyr")

```

# Start with DADA2 pipeline, create ASVs from fastq files
## set path

```{r set path, echo=F}

path_fastq <- "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/samples"
list.files(path_fastq) #listing all files in the path
fns <- sort(list.files(path_fastq, pattern=".fq", full.names = TRUE)) #selection of all fastq files
sample.names <- sapply(strsplit(basename(fns), "_"), `[`, 2)
print(sample.names)

```

## visualize sequence quality 

```{r sequence quality, echo=F}

plotQualityProfile(fns[1:6])
plotQualityProfile(fns[7:12])
plotQualityProfile(fns[13:18])
plotQualityProfile(fns[19:24])
plotQualityProfile(fns[25:30])
plotQualityProfile(fns[31:36])
plotQualityProfile(fns[37:42])
plotQualityProfile(fns[43:48])

```

## length distribution of the original sequences 

```{r original length distribution, echo=F}

#inspect length distribution of sequences to determine how to filter the sequences
length_fns <- lapply(path_fastq, function(fn) nchar(getSequences(fn)))
length <- do.call(c, length_fns)
hist(length, 100) #looks good, majority between 1400 and 1700
mean(length); median(length) 

```

## filter and trim original sequences 

```{r filter + trim, echo=F}

#place filtered files in filtered/ subdirectory 
filt_path <- file.path(path_fastq, "filtered") 
filt_fns <- file.path(path_fastq, "filtered", paste0(sample.names, "_filt.fq.gz"))
names(filt_fns) <- sample.names

#filter from 1400 to 1650
filt_out <- filterAndTrim(fns, filt_fns, minQ=2, minLen=1450, maxLen=1650, maxN=0, rm.phix=FALSE, maxEE=3, multithread = FALSE) #we retain only reads with fewer than three exp. errors 
print(filt_out)

#csv file containing number of input and output/filtered sequences
write.csv(filt_out, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/filtered_seq.csv") 

```

## length distribution of filtered sequences

```{r length distribution, echo=F}

# Inspect length distribution of filtered sequences

length_fns_filt <- lapply(filt_path, function(fn) nchar(getSequences(fn)))
length_filt <- do.call(c, length_fns_filt)
hist(length_filt, 100)
mean(length_filt); median(length_filt)
#1529.076
#1529

```

## dereplication of identical reads

```{r dereplication, echo=F}

#dereplication combines all identical sequencing reads into into “unique sequences” with a corresponding “abundance”: the number of reads with that unique sequence
#collapsing together all reads that encode the same sequence

#collapse identical sequences
derep <- derepFastq(filt_fns, verbose=TRUE)
names(derep) <- sample.names #name the derep-class objects by the sample names

```

## error rates

```{r error rates, echo=F}

#error rates for each possible transition 
set.seed(100)
err <- learnErrors(derep, errorEstimationFunction = PacBioErrfun, BAND_SIZE = 32, verbose = TRUE)  
#103155500 total bases in 67095 reads from 12 samples will be used for learning the error rates.

plotErrors(err)
plotErrors(err, nominalQ=TRUE) #red line shows the error rates, the estimated error rates (black line) are a good fit to the observed rates (points), error rates should drop with the increased quality 

```

## denoise 

```{r denoise, echo=F}

#infer the sequence variants in each sample
#separate true biological sequences from errors
dada_dd <- dada(derep, err = err, BAND_SIZE = 32, multithread=TRUE, verbose = TRUE, pool = "pseudo") #number of true sequences
dada_dd[[1]] #check only the first sample

```

# Construct sequence variant (ASV) table, a higher-resolution version of the OTU table produced by traditional methods

```{r ASV table, echo=F}

seqtab <- makeSequenceTable(dada_dd); dim(seqtab) #48 972
table(nchar(getSequences(seqtab))) #distibution of sequence lengths 
#shortest 1468, longest 1634

saveRDS(seqtab, "all_seq.rds")

#extract original sequences, belonging to 972 ASVs

asv_seqs_original <- colnames(seqtab)
asv_headers_original <- vector(dim(seqtab)[2], mode = "character")
for (i in 1:dim(seqtab)[2]) {
  asv_headers_original[i] <- paste(">ASV", i, sep = "_")
}
asv_fasta_original <- c(rbind(asv_headers_original, asv_seqs_original))
write(asv_fasta_original, "ASVs_original.fasta")

```

## remove chimeras 

```{r remove chimeras, echo=F}

seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim) 
#48 962
#chimeras removed 
#seqtab.nochim file is the final file, samples and sequences that represents ASVs

#what proportion of the sequence variants are chimeras
sum(seqtab.nochim)/sum(seqtab) 
#0.9958734
1-sum(seqtab.nochim)/sum(seqtab)  #0.004126584

```

## summary, track retained sequences

```{r summarize, echo=F}

getN <- function(x) sum(getUniques(x))
track <- cbind(filt_out, sapply(dada_dd, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoised", "nonchim")
rownames(track) <- sample.names
print(track)
class(track) #it is a matrix

#save summary 
write.csv(track, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/summary_seq.csv")

#plot the distribution of sequence length 
plot(sort(unname(rowSums(seqtab.nochim))))

#species richness
rarecurve(seqtab.nochim)

```

# Export fasta file 

```{r fasta files, echo=F}

#give our sequence headers more manageable names (ASV_1, ASV_2...) and write out a FASTA of our ASV seqs
asv_seqs <- colnames(seqtab.nochim)
asv_headers <- vector(dim(seqtab.nochim)[2], mode = "character")
for (i in 1:dim(seqtab.nochim)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep = "_")
}
asv_fasta <- c(rbind(asv_headers, asv_seqs))

#export fasta file with all 962 ASVs

write(asv_fasta, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/ASVs_processed.fasta")

```

# Export count table

```{r count table, echo=F}

asv_tab <- (seqtab.nochim)
colnames(asv_tab) <- sub(">", "", asv_headers)

write.table(asv_tab, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/ASVs_counts.csv", sep = ",", quote = FALSE, col.names = NA)

#matrix: rows corresponding to samples, columns with ASV, the value of each entry is the number of times that ASV was observed in the sample

```

# Taxonomy with UNITE 2020

```{r}

set.seed(NULL) #return to the original state by unsetting the seed
set.seed(62)

#take UNITE database

taxonomy_unite <- assignTaxonomy(seqtab.nochim, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/reference databases/sh_general_release_dynamic_04.02.2020_dev.fasta", multithread=TRUE, tryRC = TRUE)

tax_unite2020 <- taxonomy_unite # Removing sequence rownames for display only
rownames(tax_unite2020) <- NULL
rownames(tax_unite2020) <- getSequences(asv_tab)
head(tax_unite2020)

write.csv(tax_unite2020, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/tax_unite2020.csv")

write.csv(taxonomy_unite, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/tax_unite2020_seq.csv")

```

# Create phyloseq object with 962 ASV, based on UNITE taxonomy

```{r create phyloseq, echo=F}

#OTU table
OTU = otu_table(asv_tab, taxa_are_rows = FALSE)
#962 taxa, 48 samples
OTUr = transform_sample_counts(OTU, function(x) x/sum(x))

#TAX table from UNITE

taxa_unite <- read.csv("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/tax_unite2020.csv", sep = ",", row.names = 1)

tax_unite_table = tax_table(as.matrix(taxa_unite))
TAX = tax_table(tax_unite_table)

#META file 
meta_file = read.csv("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/Meta files AMF/meta_DOK.csv", sep=",", row.names=1) 
meta_file = sample_data(data.frame(meta_file))

#create phyloseq

physeq_unite = phyloseq(OTU, TAX, meta_file)
physeq_unite
#OTU_table; 962 taxa, 48 samples
#tax_table; 962 taxa, by 7 taxonomic ranks
#meta_file; 48 samples, 24 variables 

physeq_unite_rel = phyloseq(OTUr, tax_unite, meta_file)

sort(sample_sums(physeq_unite), decreasing = F) 
sum(otu_table(physeq_unite)) #185825 sequencing reads in all samples

#add tree
library("ape"); packageVersion("ape") #5.3
unite_tree = rtree(ntaxa(physeq_unite), rooted=TRUE, tip.label=taxa_names(physeq_unite))
plot((unite_tree))

#add unite tree to phyloseq object 
physeq_unite = merge_phyloseq(physeq_unite, unite_tree)
physeq_unite

#plot relative abundance 
plot_bar(tax_glom(physeq_unite_rel, "Phylum"), fill="Phylum")
plot_bar(physeq_unite_rel, x = "treatment:farming_system", fill = "Class")
plot_bar(tax_glom(physeq_unite_rel, "Phylum"), fill="Phylum") + facet_grid(.~treatment *time, scale="free_x", space = "free_x")

```
**Initial phyloseq object with 962 ASVs in 48 samples is created. Total number of sequences is 185,825.**

## filter phyloseq object

```{r filter phyloseq object, echo=F}

#remove non-AMF 
physeq_unite_Filtered <- subset_taxa(physeq_unite, !Phylum %in% c ("p__Basidiomycota", "p__Ascomycota")) #we removed non-AMF and now we have 955 ASVs
physeq_unite_Filtered

physeq_unite_Filtered_rel <- subset_taxa(physeq_unite_rel, !Phylum %in% c ("p__Basidiomycota", "p__Ascomycota"))

plot_bar(tax_glom(physeq_unite_Filtered_rel, "Phylum"), fill="Phylum")

sum(sample_sums(physeq_unite)) #185825
sum(sample_sums(physeq_unite_Filtered)) #185364
sort(sample_sums(physeq_unite_Filtered), decreasing = F) #we will discard samples with less than 500 reads 

```

## phyloseq object with only AMF (Glomeromycota)

```{r AMF phyloseq, echo=F}

physeq_AMF <- physeq_unite_Filtered

#compute prevalence of each feature, store as data.frame
prevdf=apply(X=otu_table(physeq_AMF), 
             MARGIN=ifelse(taxa_are_rows(physeq_AMF), yes=1, no=2),
             FUN=function(x){sum(x>0)})

#add taxonomy and total read frames to this data.frame
prevdf=data.frame(Prevalence=prevdf,
                  TotalAbundance=taxa_sums(physeq_AMF), tax_table(physeq_AMF))

min(sample_sums(physeq_AMF)) #404
max(sample_sums(physeq_AMF)) #11426
median(sample_sums(physeq_AMF)) #3414.5
mean(sample_sums(physeq_AMF)) #3861.75

#how many taxa are there? 
unique(ntaxa(physeq_AMF)) #955

#save phyloseq object
saveRDS(physeq_AMF, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/phyloseq_AMF.rds")

```
**955 ASVs in Glomeromycota phylum**

## rarefaction curves

```{r rarefaction curves, echo=F}

rarecurve(otu_table(physeq_AMF), step=100, label = FALSE, xlab="Number of sequences",  ylab="Number of ASVs")

barplot(sort(sample_sums(physeq_AMF)),
        ylab= "Number of reads", xlab="Sample ID")

```
**To determine the completeness of our sampling efforts, we plotted rarefaction curves for the 48 experimental units analyzed in this study using the “rarecurve” function from the package vegan in R. ASV richness reached clear asymptotes for all experimental units, indicating that there was sufficient sequencing depth to recover the majority of AMF ASVs.**

#Convert from phyloseq to vegan or from vegan to phyloseq

```{r}

# convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

```

# COMMUNITY COMPOSITION 
#### here I will explore the community composition on different taxonomic levels, ASV, species and genus level

#ASV level 

```{r community composition ASV level, echo=F}

## ASV level 

#remove samples with less than 500 reads 
physeq_unite_Filtered_ASV <- prune_samples(sample_sums(physeq_AMF) >=500, physeq_AMF)

sort(sample_sums(physeq_unite_Filtered_ASV), decreasing = F) #we keep 46 samples

#rarefy on the minimal number of reads

unite_ASV_rare <- rarefy_even_depth(physeq_unite_Filtered_ASV, sample.size = min(sample_sums(physeq_unite_Filtered_ASV)), 6020, replace=F, verbose = T) # rarefication on the lowest number of reads and as a result with 941 ASVs

sort(sample_sums(unite_ASV_rare))

```
**After rarefaction we have 46 samples, raraefied on 622 reads per sample.**
 
##plot ordination on ASV level 

```{r plot ordination ASV level, echo=F}

#PCoA

ASV_unite_pcoa <- phyloseq::ordinate(unite_ASV_rare, method = "PCoA") # building pcoa

plot_ordination(unite_ASV_rare , ASV_unite_pcoa, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw()
plot_ordination(unite_ASV_rare , ASV_unite_pcoa, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw()

plot_ordination(unite_ASV_rare, ASV_unite_pcoa, type="samples", color="farming_system",label="sample_ID")+
               geom_text(mapping = aes(label = sample_ID), size =2, vjust = 1.5) +
               geom_point(size=5) +
               ggtitle("ASV level") +
               facet_grid(~time)

```
**Very difficult to extract the information from the ordination plot.**

###PERMANOVA with BC on ASV level 

```{r permanova ASV level, echo=F}

## Perform PERMANOVA testing for main effects (system, treatment, time) and interactions between time&system&treatment with block in strata

#TSS normalization
unite_ASV_rel <- transform_sample_counts(unite_ASV_rare, function(x) x/sum(x))
bray_uniteASV <- phyloseq::distance(unite_ASV_rel, method = "bray")

meta.ASV <- as(sample_data(unite_ASV_rel), "data.frame")

set.seed(1546)
perm.ASV <- adonis2(bray_uniteASV ~ treatment+time+farming_system+
                    farming_system:treatment+
                    treatment:time+
                    time:farming_system,
                    strata=meta.ASV$block,
                    data=meta.ASV,permutations = 999)

perm.ASV

#save output for paper
permanova_ASV <- as.data.frame(perm.ASV)
permanova_ASV <- permanova_ASV[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(permanova_ASV, "permanova_ASV.txt", sep="\t",quote=F)

```
**Time effect**

## plot ordination UniFrac on ASV level 

```{r plot unifrac ASV , echo=F}

#weighted Unifrac plot
ord_pcoa_wuni <- ordinate(unite_ASV_rare, "PCoA", "unifrac", weighted = TRUE)
barplot(ord_pcoa_wuni$values$Eigenvalues[1:10])

plot_ordination(unite_ASV_rare, ord_pcoa_wuni, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw() + ggtitle("Weighted UniFrac")
plot_ordination(unite_ASV_rare, ord_pcoa_wuni, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw() + ggtitle("Weighted UniFrac")

#unweighted Unifrac plot
ord_pcoa_uni <- ordinate(unite_ASV_rare, "PCoA", "unifrac", weighted = FALSE)
barplot(ord_pcoa_uni$values$Eigenvalues[1:10])

plot_ordination(unite_ASV_rare, ord_pcoa_uni, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw() + ggtitle("Unweighted UniFrac")
plot_ordination(unite_ASV_rare, ord_pcoa_uni, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw() + ggtitle("Unweighted UniFrac")

```

## weighted Unifrac distance on ASV level 

```{r weighted unifrac distance, echo=F}

wuni_asv <- UniFrac(unite_ASV_rare, weighted = TRUE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.wuni_asv <- as(sample_data(unite_ASV_rare), "data.frame")

set.seed(1420)

perm_mod_wuni_asv <- adonis2(wuni_asv ~ treatment+time+farming_system+
                                       farming_system:treatment+
                                       treatment:time+
                                       time:farming_system,
                                       strata=meta.wuni_asv$block,
                                       data = meta.wuni_asv,permutations = 999)
perm_mod_wuni_asv

```
**Weighted UniFrac distance. No effect of any of the factors for weighted UniFrac AMF community composition on ASV level.**

## unweighted Unifrac distance on ASV level

```{r unweighted unifrac distance, echo=F}

uni_asv <- UniFrac(unite_ASV_rare, weighted = FALSE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.uni_asv <- as(sample_data(unite_ASV_rare), "data.frame")

set.seed(1420)

perm_mod_uni_asv <- adonis2(uni_asv ~ treatment+time+farming_system+
                                     farming_system:treatment+
                                     treatment:time+
                                     time:farming_system,
                                     strata=meta.uni_asv$block,
                                     data = meta.uni_asv,permutations = 999)
perm_mod_uni_asv
```
**Time and system slightly affected the unweighted UniFrac of AMF composition, ASV level.**

#SPECIES level 

```{r species level, echo=F}

physeq_AMF_spec = tax_glom(physeq_AMF, "Species", NArm = T)
unique(tax_table(physeq_AMF_spec)[,7])

sum(sample_sums(physeq_AMF_spec)) #84542
sort(sample_sums(physeq_AMF_spec), decreasing = F)

#keep samples with more than 500 reads 
physeq_AMF_spec_500 <- prune_samples(sample_sums(physeq_AMF_spec) >=500, physeq_AMF_spec)
sort(sample_sums(physeq_AMF_spec_500))

#rarefaction
unite_spec_rare <- rarefy_even_depth(physeq_AMF_spec_500, sample.size = min(sample_sums(physeq_AMF_spec_500)), 6020, replace=F, verbose = T) # rarefication on the lowest number of reads 

```
**Agglomeration on species level and rarefaction on the lowest number of sequencing reads, we have 39 samples.**

##plot ordination on species level

```{r plot ordination on species level,echo=F}

#plot PCoA
spec_unite_pcoa <- phyloseq::ordinate(unite_spec_rare, method = "PCoA") # building pcoa

plot_ordination(unite_spec_rare, spec_unite_pcoa, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw()
plot_ordination(unite_spec_rare, spec_unite_pcoa, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw()

#NMDS
spec_unite_nmds <- phyloseq::ordinate(unite_spec_rare, method = "NMDS") 
spec_unite_nmds$stress #0.2280184

plot_ordination(unite_spec_rare, spec_unite_nmds, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw() 
plot_ordination(unite_spec_rare, spec_unite_nmds, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw() 

```
**NMDS and PCoA on species level**

###PERMANOVA with BC on species level 

```{r permanova on species level, echo=F}

unite_spec_rel <- transform_sample_counts(unite_spec_rare, function(x) x/sum(x))
bray_unite_spec <- phyloseq::distance(unite_spec_rel, method = "bray")

meta.unite_spec <- as(sample_data(unite_spec_rare), "data.frame")

set.seed(1546)
perm.spec <- adonis2(bray_unite_spec ~ treatment+time+farming_system+
                    farming_system:treatment+
                    treatment:time+
                    time:farming_system,
                    strata=meta.unite_spec$block,
                    data = meta.unite_spec,permutations = 999)
perm.spec
```
**Effect of time on species level on AMF composition.**

#GENUS level 

```{r community composition genus level, echo=F}

## agglomerate on genus level 

length(get_taxa_unique(physeq_AMF, taxonomic.rank = "Genus"))

# agglomerate taxa at the genus level (combine all with the same name) and remove all taxa without genus level assignment

physeq_AMF_genus = tax_glom(physeq_AMF, "Genus", NArm = T)
physeq_AMF_genus #agglomerated on the genus level

sum(sample_sums(physeq_AMF_genus)) #128352

unique(tax_table(physeq_AMF_genus)[,6]) #12

sort(sample_sums(physeq_AMF_genus), decreasing = F) #we will remove 3 samples

#keep samples with more than 500 reads 
physeq_AMF_genus_500 <- prune_samples(sample_sums(physeq_AMF_genus) >=500, physeq_AMF_genus)

sort(sample_sums(physeq_AMF_genus_500))
plot_bar((physeq_AMF_genus_500))
unique(ntaxa(physeq_AMF_genus_500)) #12
any(taxa_sums(physeq_AMF_genus_500) == 0)

#rank-abundance barplot
par(mar = c(5, 10, 4, 2) + 0.1) # make more room on bottom margin
N <- 12
barplot(sort(taxa_sums(physeq_AMF_genus_500), TRUE)[1:N]/nsamples(physeq_AMF_genus_500), las=2)

#rarefaction
unite_genus_rare <- rarefy_even_depth(physeq_AMF_genus_500, sample.size = min(sample_sums(physeq_AMF_genus_500)), 6020, replace=F, verbose = T) # rarefication on the lowest number of reads 

sort(sample_sums(unite_genus_rare))

```
**Agglomeration on genus level, after rarefaction on 531 reads we have 45 samples.**

##plot ordination on genus level

```{r plot ordination genus level, echo=F}

#plot PCoA
genus_unite_pcoa <- phyloseq::ordinate(unite_genus_rare, method = "PCoA") # building pcoa

plot_ordination(unite_genus_rare, genus_unite_pcoa, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw()
plot_ordination(unite_genus_rare, genus_unite_pcoa, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw()

plot_ordination(unite_genus_rare, genus_unite_pcoa, type="samples", color="farming_system",label="sample_ID")+
               geom_text(mapping = aes(label = sample_ID), size =2, vjust = 1.5) +
               geom_point(size=5) +
               ggtitle("Agglomerated on genus level") +
               facet_grid(~time)

#plot NMDS
set.seed(0902)
genus_unite_nmds <- phyloseq::ordinate(unite_genus_rare, method = "NMDS") 
genus_unite_nmds$stress #0.1902684


plot_ordination(unite_genus_rare, genus_unite_nmds, color = "treatment", shape = "time") + geom_point(size = 3) + theme_bw() 
plot_ordination(unite_genus_rare, genus_unite_nmds, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw() 

```

# NMDS ordination (genus level)

```{r, echo=F}

#transfer OTU table from phyloseq into a vegan object
vegan_AMF_genus<-psotu2veg(unite_genus_rare)

#transfer sample data from phyloseq into a vegan object
veganSample_AMF_genus<-pssd2veg(unite_genus_rare)

#calculate the distance matrix
set.seed(0902)
nmds_ord <- metaMDS(vegan_AMF_genus, distance = "bray", k=2, trymax=1000, autotransform=T, permutations = 999)
nmds_ord$stress

plot_ordination(unite_genus_rare, nmds_ord, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw()

#stress is goodness of fit 
goodness(nmds_ord)
stressplot(nmds_ord)
 
#this function draws NMDS ordination diagram with sites (=my samples)
plot(nmds_ord, display = 'sites', type = 't', main = 'Goodness of fit')

```

#envfit 

```{r, echo=F}

#envfit with metadata 
nmds_meta <- as(sample_data(unite_genus_rel), "data.frame")
env_nmds <- nmds_meta[,5:24]

data.scores_nmds = as.data.frame(scores(nmds_ord, display = 'sites')) #save NMDS to dataframe

#add grouping variable to dataframe
data.scores_nmds <- cbind(as.data.frame(data.scores_nmds), farming_system = env_nmds$farming_system, 
                          treatment = env_nmds$treatment)
head(data.scores_nmds)

```

## calculate envfit for ndms ordination
## investigate the soil parameters which may be driving the distribution pattern
```{r, echo=F}

#calculate envfit
set.seed(0935)
env.fit <- envfit(scores(nmds_ord), env_nmds, permutations = 999, na.rm = TRUE, strata=nmds_meta$block)
env.fit

#extract scores from environment
env.scores <- as.data.frame(scores(env.fit, display = "vectors"))
env.scores <- cbind(env.scores, env.variables = rownames(env.scores), pval = env.fit$vectors$pvals)
head(env.scores)
sig.env <- subset(env.scores, pval<=0.05) #extract only significant soil parameters
head(sig.env)

#envfit write table for paper

envfit_nmds <- data.frame((env.fit$vectors)$r, (env.fit$vectors)$pvals)
write.table(envfit_nmds, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/envfit_nmds.txt", sep = "\t", quote = FALSE, row.names = T, col.names=T)

#vectors as continuous, factors as categorical
#vectors are soil parameters and factors (centroids) are treatment + farming system
en_coord_cont = as.data.frame(scores(env.fit, "vectors")) * ordiArrowMul(env.fit)
en_coord_cat = as.data.frame(scores(env.fit, "factors")) * ordiArrowMul(env.fit)

```

## final NMDS plot 

```{r NMDS plot on genus level, echo=F}
#plot with significant environmental parameters

nmds_final <- ggplot() + geom_point(data = data.scores_nmds, aes(x = NMDS1, y = NMDS2, colour = farming_system), size = 5, alpha = 0.5) + theme_bw() + scale_colour_manual("Farming system", values=c("grey70", "grey25")) + theme(axis.title = element_text(size = 12, colour = "black"), axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), legend.title = element_text(size = 11, colour = "black"), legend.text = element_text(size = 10, colour = "black")) + labs(colour = "farming_system") + annotate("text", x=1.0, y=-0.7, size=5,label= paste("Stress =", round(nmds_ord$stress, digits=2))) + geom_segment(data = sig.env, aes(x = 0, xend=NMDS1, y=0, yend=NMDS2), arrow = arrow(length = unit(0.25, "cm")), colour = "black", lwd=0.3) + ggrepel::geom_text_repel(data = sig.env, aes(x=NMDS1, y=NMDS2, label = env.variables), cex = 4, color="black", direction = "both", segment.size = 0.25) + geom_point(data = en_coord_cat, aes(x = NMDS1, y = NMDS2), shape = "diamond", size = 5, alpha = 0.8, colour = "black") + geom_text(data = en_coord_cat, aes(x = NMDS1, y = NMDS2+0.04), label = row.names(en_coord_cat), colour = "black", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

print(nmds_final)
ggsave("results_dada2/nmds_final.pdf", width = 40, height = 20, units = "cm")
ggsave("results_dada2/nmds_final.tiff", width = 40, height = 20, units = "cm")

```

###PERMANOVA with BC on genus level 

```{r permanova on genus level, echo=F}

unite_genus_rel <- transform_sample_counts(unite_genus_rare, function(x) x/sum(x))
bray_unite_genus <- phyloseq::distance(unite_genus_rel, method = "bray")

meta.unite_genus <- as(sample_data(unite_genus_rare), "data.frame")

set.seed(1546)
perm.genus <- adonis2(bray_unite_genus ~ treatment+time+farming_system+
                    farming_system:treatment+
                    treatment:time+
                    time:farming_system,
                    strata=meta.unite_genus$block,
                    data = meta.unite_genus,permutations = 999)
perm.genus

#effect of time and farming system on genus composition with bray distance 
#we should highlight with constraint method

#save output for paper
permanova_genus <- as.data.frame(perm.genus)
permanova_genus <- permanova_genus[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(permanova_genus, "permanova_genus.txt", sep="\t",quote=F)

```
**Effect of the time and the farming system on genus composition with BC distance.We should highlight these effects with constrained method.**

##weighted Unifrac distance on genus level

```{r weighted unifrac distance genus, echo=F}

wuni_genus <- UniFrac(unite_genus_rare, weighted = TRUE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.wuni_genus <- as(sample_data(unite_genus_rare), "data.frame")

set.seed(1420)
perm_mod_wuni_genus <- adonis2(wuni_genus ~ treatment+time+farming_system+
                                       farming_system:treatment+
                                       treatment:time+
                                       time:farming_system,
                                       strata=meta.wuni_genus$block,
                                       data = meta.wuni_genus,permutations = 999)
perm_mod_wuni_genus

```
**Time significance for weighted Unifrac on genus level AMF composition.**

##unweighted Unifrac distance on genus level

```{r unweighted unifrac distance genus, echo=F}

uni_genus <- UniFrac(unite_genus_rare, weighted = FALSE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.uni_genus <- as(sample_data(unite_genus_rare), "data.frame")

set.seed(1420)
perm_mod_uni_genus <- adonis2(uni_genus ~ treatment+time+farming_system+
                                     farming_system:treatment+
                                     treatment:time+
                                     time:farming_system,
                                     strata=meta.uni_genus$block,
                                     data = meta.uni_genus,permutations = 999)
perm_mod_uni_genus
```
**Slight interaction between treatment and time for genus AMF composition with unweighted UniFrac.**

# Constrained analysis with db-RDA, to highlight effects
###Plot only the variation in your data that explains changes in community composition

```{r plot db-RDA, echo=F}

set.seed(1120)

#add volumetric soil water content (VWC)

veganSample_AMF_genus$VWC<-veganSample_AMF_genus$WC*veganSample_AMF_genus$BD
summary(veganSample_AMF_genus$VWC)

#create the distance matrix
dbrda <- dbrda(vegan_AMF_genus ~ farming_system+time+Condition(block),veganSample_AMF_genus,dist="bray")

anova(dbrda, by="axis")
anova(dbrda, by="term", permutations = 999)

#calculate amount of variation explained by the axis
b_xlabel<-paste0("db-RDA1"," ","[", round(((dbrda$CCA$eig[1]/dbrda$CA$tot)*100),1), "%]" )
b_ylabel<-paste0("db-RDA2"," ","[", round(((dbrda$CCA$eig[2]/dbrda$CA$tot)*100),1), "%]" )

#use ggplot to plot the constraint ordination
library(devtools)
# devtools::install_github("gavinsimpson/ggvegan")
library(ggvegan)

#fortify: transfer into data frame that can be used for ggplot
scrs <- fortify(dbrda)

#select site (-> Samples)
sites <- subset(scrs, subset = Score == "sites") 

#species <- subset(scrs, subset = Score == "species")

(Time<-veganSample_AMF_genus$time) #choose group variable here
(System<-veganSample_AMF_genus$farming_system) #choose shape variable
df <- cbind(sites, Time , System) ## add on the group variable

#make the plot
dbrda_plot<-ggplot(df, aes(x = dbRDA1, y = dbRDA2)) + 
  geom_hline(yintercept=0, linetype="dotted") +
  geom_vline(xintercept=0, linetype="dotted") +
  xlab(b_xlabel)+
  ylab(b_ylabel)+
  geom_point( aes(color= System, shape=Time),
              size = 8,
              alpha = 0.5,
              show.legend = T)+
  scale_shape_manual(values=c(16,17,18), name="")+
  scale_color_manual(values=c("black","grey"), labels=c("BIODYN", "CONMIN"), name="")+
  ggtitle("AMF")+
  theme(legend.position = "bottom", 
        legend.text=element_text(size=14),
        legend.title=element_blank(),
        plot.margin = unit(c(0,3, 3,3), "mm"))

print(dbrda_plot)

```

## add variables to constrained db-RDA
### Which environmental variables correlate with the ordination projection?

```{r add soil variables, echo=F}

names(veganSample_AMF_genus)

b_treatments_env <- veganSample_AMF_genus[,c("pH", "WC", "BD", "C.N", "SOC", "total.N","weed.cover", "PO4_P", "total.P","root.DW", "WHC", "NLFA", "N.roots", "C.roots", "crop.biomass", "plant.height", "N.shoots", "C.shoots", "VWC")]

#scale variables (z-transform) because they are measured on very different scales

b_treatments_env$VWC<-scale(b_treatments_env$VWC)
b_treatments_env$pH<-scale(b_treatments_env$pH)
b_treatments_env$C.N<-scale(b_treatments_env$C.N)
b_treatments_env$total.N<-scale(b_treatments_env$total.N)
b_treatments_env$SOC<-scale(b_treatments_env$SOC)
b_treatments_env$weed.cover<-scale(b_treatments_env$weed.cover)
b_treatments_env$total.P<-scale(b_treatments_env$total.P)
b_treatments_env$PO4_P<-scale(b_treatments_env$PO4_P)
b_treatments_env$root.DW<-scale(b_treatments_env$root.DW)
b_treatments_env$WHC<-scale(b_treatments_env$WHC)
b_treatments_env$BD<-scale(b_treatments_env$BD)
b_treatments_env$plant.height<-scale(b_treatments_env$plant.height)
b_treatments_env$N.shoots<-scale(b_treatments_env$N.shoots)
b_treatments_env$C.shoots<-scale(b_treatments_env$C.shoots)
b_treatments_env$NLFA<-scale(b_treatments_env$NLFA)
b_treatments_env$crop.biomass<-scale(b_treatments_env$crop.biomass)
b_treatments_env$WC<-scale(b_treatments_env$WC)
b_treatments_env$N.roots<-scale(b_treatments_env$N.roots)
b_treatments_env$C.roots<-scale(b_treatments_env$C.roots)

#Run the actual envfit function, set Strata, to account for the blocking in the DOK
#use linear combination of scores (display="lc")

set.seed(1722)
b_vec.sp <- envfit(dbrda, b_treatments_env, strata=veganSample_AMF_genus$block,na.rm = TRUE, display="lc")

#if you want to save results in a df and export run:

#add arrows to the plot
#now you can decided if you want to plot vectors based on
#p-values or based on r2 values. I do the latter.
#shortcutting ef$vectors
b_A <- as.list(b_vec.sp$vectors)
(b_A)

#creating the dataframe
b_r2<-as.data.frame(b_A$r)
b_r2

#scale arrows for latter plotting according to the correlation stength.
b_arrows<-as.data.frame(b_A$arrows*4.5*(b_A$r))

b_C<-cbind(b_arrows, b_A$r)
b_C

#subset the correlations with a r>0.4
b_vec.sp.df<-droplevels(subset(b_C,b_A$r>=0.4))
b_vec.sp.df <- cbind(b_vec.sp.df, Species = rownames(b_vec.sp.df))
b_vec.sp.df


## Add environmental variables to the plot
dbrda_plot_env<-dbrda_plot+
  geom_segment(data = b_vec.sp.df,
               aes(x = 0, xend = dbRDA1,
                   y = 0, yend = dbRDA2),
               arrow = arrow(length = unit(2.5, "mm")),
               colour = "black",
               alpha=1,
               size=0.6,
               lineend ="round",
               linejoin="round")+
  geom_text(data=b_vec.sp.df,
            aes(x=dbRDA1,
                y=dbRDA2,
                label=Species),
            color="black",
            alpha=0.8,
            size=5,
            vjust=0,
            hjust=0,
            check_overlap = F,
            position=position_jitter(width=0.2,height=0.1))

dbrda_plot_env

```

#CAP analysis 

```{cap genus level, echo=F}


```



# ALPHA diversity indices

```{r alpha_div, echo=F}

#the alpha diversity metrics were calculated based on samples rarefied to the lowest number of reads, calcualted per sample
#alpha diversity estimates on rarefied data
#we calculate Observed and Shannon for each sample

alpha_div <- data.frame(estimate_richness(unite_ASV_rare, split=TRUE, measures = c("Observed", "Shannon")), sample_data(unite_ASV_rare))
View(alpha_div)

plot_richness(unite_ASV_rare, x="treatment", color="farming_system", measures=c("Observed","Shannon")) + stat_boxplot(geom = "errorbar") + geom_boxplot() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + scale_color_manual(values=wes_palette(n=2, name="Cavalcanti1")) + xlab("Treatment") + ggtitle("AMF alpha diversity") + ylab("Alpha metrics")

write.csv(alpha_div, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/alpha_div.csv")

```
**Observed ASV richness and Shannon diversity index are calculated.**

## prepare data for BRMS model

```{r prepare data for BRMS model, echo=F}

#import data
head(alpha_div)

data_brms <- alpha_div
head(data_brms)
str(data_brms)

#load libraries
library(brms); packageVersion("brms") #2.13.0
library(shinystan); packageVersion("shinystan") #2.5.0

```

## BRMS model for observed richness 

```{r BRMS for observed richness, echo=F}

observed_brms <-brm(formula = (Observed)~0+treatment*farming_system+(1|block/plot)+(1|time),
            data = data_brms,
            family="gaussian",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, max_treedepth = 15))

# Save an object to a file
saveRDS(observed_brms, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/observed_brms.rds")

#reload it so you don't have to run it again
observed_brms <- readRDS("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/observed_brms.rds")

#check model fit 
yfit_observed <- data_brms$Observed
yrep_observed <- posterior_predict(observed_brms, draws = length(yfit_observed))

launch_shinystan(observed_brms)

#explore results
summary(observed_brms)
bayes_R2(observed_brms)

#marginal effects
marginal_effects(observed_brms)

```
**BRMS model for Observed richness is created.**

## plot BRMS observed richness

```{r plot brms for observed richness, echo=F}

#we can plot the summary of the posterior distribution.
#make a data frame that contains the parameters of interest
#here: all of them

data_brms$farming_system <- as.factor(data_brms$farming_system)
data_brms$treatment <- as.factor(data_brms$treatment)

newdat <- expand.grid(
  farming_system = factor(c("BIODYN", "CONMIN"), levels=levels(data_brms$farming_system)),
  treatment = factor(c("C","R", "RC"),levels=levels(data_brms$treatment)))
head(newdat)

#use the function "fitted" from brms to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_observed = fitted(observed_brms, newdata = newdat, robust=T, re_formula = NA,summary = TRUE, probs = c(0.025, 0.975))  
fit_observed

# no colnames, I add them manually
fitdf_observed_plot = cbind(newdat, fit_observed)
fitdf_observed_plot

#prepare and extract results table
fitdf_observed<-fitdf_observed_plot[, c("farming_system","treatment","Estimate","Q2.5","Q97.5")]
write.table(fitdf_observed,"/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/observed_alphadiv_brms.txt",sep="\t",quote=F)

#change order of treatment in fitted values
fitdf_observed_plot$treatment<-factor(fitdf_observed_plot$treatment, levels=c("R", "RC", "C"))

#this is the dataframe i will use for plotting
pos.dodge <- position_dodge(0.7) # move them .7 to the left and right

#make the final plot, without raw data
fitted_observed<-ggplot(data = fitdf_observed_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values =  c("palegreen4",
                                 "steelblue2",
                                 "tomato2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  ylab("Observed ASV")+theme_bw()

print(fitted_observed)
ggsave("results_dada2/observed_brms.tiff", width = 18, height = 10, units = "cm")

#get location of the means including credible intervals from here:
fitdf_observed_plot<-as.data.frame(fitdf_observed_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_observed = as.data.frame(fitted(observed_brms, newdata=newdat,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat$farming_system, newdat$treatment))

colnames(fit_observed)<-colnames
names(fit_observed)

#calculate the mean differences between systems across the roofs
BIODYN_brms_obs<-(fit_observed$BIODYN.C+fit_observed$BIODYN.RC+fit_observed$BIODYN.R)/3
mean(BIODYN_brms_obs)
sd(BIODYN_brms_obs)
CONMIN_brms_obs<-(fit_observed$CONMIN.C+fit_observed$CONMIN.RC+fit_observed$CONMIN.R)/3
mean(CONMIN_brms_obs)
sd(CONMIN_brms_obs)

diff_brms_obs<-(BIODYN_brms_obs-CONMIN_brms_obs)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_obs, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_obs)

```

## BRMS model for Shannon index 

```{r BRMS for observed ASVs richness, echo=F}

shannon_brms <-brm(formula = (Shannon)~0+treatment*farming_system+(1|block/plot/treatment) + (1|time),
            data = data_brms,
            family="gaussian",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, max_treedepth = 15))

# Save an object to a file
saveRDS(shannon_brms, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/shannon_brms.rds")

#reload it so you don't have to run it again
shannon_brms <- readRDS("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/shannon_brms.rds")

#check model fit 
yfit_shannon <- data_brms$Shannon
yrep_shannon <- posterior_predict(shannon_brms, draws = length(yfit_shannon))

launch_shinystan(shannon_brms)

#explore results
summary(shannon_brms)

#marginal effects
marginal_effects(shannon_brms)

```
**BRMS model for Shannon index is created.**

## plot brms for Shannon index

```{r plot brms for Shannon, echo=F}

#use the function "fitted" from brms
#to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_shannon = fitted(shannon_brms, newdata = newdat, robust=T, re_formula = NA,summary = TRUE,probs = c(0.025, 0.975))
fit_shannon

# no colnames, I add them manually
fitdf_shannon_plot = cbind(newdat, fit_shannon)
fitdf_shannon_plot

#prepare and extract results table
fitdf_shannon<-fitdf_shannon_plot[, c("farming_system","treatment","Estimate","Q2.5","Q97.5")]
write.table(fitdf_shannon,"/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/shannon_alphadiv_brms.txt",sep="\t",quote=F)

#change order of treatment in fitted values
fitdf_shannon_plot$treatment<-factor(fitdf_shannon_plot$treatment, levels=c("R", "RC", "C"))

#this is the dataframe i will use for plotting
pos.dodge <- position_dodge(0.7) # move them .7 to the left and right

#make the final plot
fitted_shannon<-ggplot(data = fitdf_shannon_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values =  c("palegreen4",
                                 "steelblue2",
                                 "tomato2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  ylab("Shannon diversity index")+theme_bw()

print(fitted_shannon)
ggsave("results_dada2/shannon_brms.tiff", width = 18, height = 10, units = "cm")

#get location of the means including credible intervals from here:
fitdf_shannon_plot<-as.data.frame(fitdf_shannon_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_shannon = as.data.frame(fitted(shannon_brms, newdata=newdat,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat$farming_system, newdat$treatment))

colnames(fit_shannon)<-colnames
names(fit_shannon)

#calculate the mean differences between systems across the roofs
BIODYN_brms_shannon<-(fit_shannon$BIODYN.C+fit_shannon$BIODYN.RC+fit_shannon$BIODYN.R)/3
mean(BIODYN_brms_shannon)
sd(BIODYN_brms_shannon)
CONMIN_brms_shannon<-(fit_shannon$CONMIN.C+fit_shannon$CONMIN.RC+fit_shannon$CONMIN.R)/3
mean(CONMIN_brms_shannon)
sd(CONMIN_brms_shannon)

diff_brms_shannon <- (BIODYN_brms_shannon-CONMIN_brms_shannon)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_shannon, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_shannon)

```

## phylogenetic diversity, calculated with Faith's PD index
Faith's PD is defined as the total branch length spanned by the tree including all species in a local community, optionally including the root node of the phylogeny.

```{r faith index, echo=F}
#The pd function returns two values for each community, Faith's PD and observed species richness (SR).

library(picante); packageVersion("picante") #1.8

faith_asvtable <- as.data.frame(unite_ASV_rare@otu_table)
faith_tree <- unite_ASV_rare@phy_tree

#check if the tree is rooted or not
faith_tree

#if yes
faith_pd <- pd(faith_asvtable, faith_tree)
head(faith_pd)

write.csv(faith_pd, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/faith_index.csv")

#add phylogenetic_diversity to alpha diversity file and to brms data
alpha_div$phylogenetic_diversity <- faith_pd$PD
data_brms$Faith <- faith_pd$PD

#Plot Faith's PD by farming system
boxplot(faith_pd$PD ~ unite_ASV_rare@sam_data$farming_system, xlab="Farming system", ylab="Faith's PD")
t.test(faith_pd$PD ~  unite_ASV_rare@sam_data$farming_system)

#Plot Faith's PD by treatment
boxplot(faith_pd$PD ~ unite_ASV_rare@sam_data$treatment, xlab="Treatment", ylab="Faith's PD") 

#Compare PD and species richness
plot(faith_pd$PD ~ faith_pd$SR, xlab = "Species richness", ylab = "Faith's PD")

#Plot PD index 
faith_plot <- ggplot(alpha_div, aes(x=treatment, y=phylogenetic_diversity, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Treatment") + ylab("Faith's PD") +  scale_x_discrete(limits=c("C","R","RC")) + geom_boxplot() + scale_fill_manual(values=c("darkgreen", "darkgoldenrod1")) + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) 

print(faith_plot)

```

## BRMS model for Faith's index

```{r}

faith_brms <-brm(formula = (Faith)~0+treatment*farming_system+(1|block/plot/treatment) + (1|time),
            data = data_brms,
            family="gaussian",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, max_treedepth = 15))

# Save an object to a file
saveRDS(faith_brms, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/faith_brms.rds")

#reload it so you don't have to run it again
faith_brms <- readRDS("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/faith_brms.rds")

#check model fit 
yfit_faith <- data_brms$Faith
yrep_faith <- posterior_predict(faith_brms, draws = length(yfit_faith))

launch_shinystan(faith_brms)

#explore results
summary(faith_brms)

#marginal effects
marginal_effects(faith_brms)

```
**BRMS model for Faith's index is created.**

## plot brms for Faith's index

```{r}

#use the function "fitted" from brms
#to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_faith = fitted(faith_brms, newdata = newdat, robust=T, re_formula = NA,summary = TRUE,probs = c(0.025, 0.975))
fit_faith

# no colnames, I add them manually
fitdf_faith_plot = cbind(newdat, fit_faith)
fitdf_faith_plot

#prepare and extract results table
fitdf_faith<-fitdf_faith_plot[, c("farming_system","treatment","Estimate","Q2.5","Q97.5")]
write.table(fitdf_faith,"/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results_dada2/faith_alphadiv_brms.txt",sep="\t",quote=F)

#change order of treatment in fitted values
fitdf_faith_plot$treatment<-factor(fitdf_faith_plot$treatment, levels=c("R", "RC", "C"))

#this is the dataframe i will use for plotting
pos.dodge <- position_dodge(0.7) # move them .7 to the left and right

#make the final plot
fitted_faith<-ggplot(data = fitdf_faith_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values =  c("palegreen4",
                                 "steelblue2",
                                 "tomato2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  ylab("Faith's phylogenetic index")+theme_bw()

print(fitted_faith)
ggsave("results_dada2/faith_brms.tiff", width = 18, height = 10, units = "cm")

#get location of the means including credible intervals from here:
fitdf_faith_plot<-as.data.frame(fitdf_faith_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_faith = as.data.frame(fitted(faith_brms, newdata=newdat,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat$farming_system, newdat$treatment))

colnames(fit_faith)<-colnames
names(fit_faith)

#calculate the mean differences between systems across the roofs
BIODYN_brms_faith<-(fit_faith$BIODYN.C+fit_faith$BIODYN.RC+fit_faith$BIODYN.R)/3
mean(BIODYN_brms_faith)
sd(BIODYN_brms_faith)
CONMIN_brms_faith<-(fit_faith$CONMIN.C+fit_faith$CONMIN.RC+fit_faith$CONMIN.R)/3
mean(CONMIN_brms_faith)
sd(CONMIN_brms_faith)

diff_brms_faith <- (BIODYN_brms_faith-CONMIN_brms_faith)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_faith, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_faith)

```

# INDICATOR SPECIES ANALYSIS
####to investigate if there are indicators of the farming system, the time of sampling and drought treatment

```{r indicator species analysis, echo=F}

library(indicspecies); packageVersion("indicspecies") #1.7.8

#Transfer Phyloseq into a vegan ASV table
ind_analysis <- psotu2veg(physeq_AMF)

#meta file for indicators 
ind_meta <- as(sample_data(physeq_AMF), "data.frame")

ind_time <- ind_meta$time
ind_treatment <- ind_meta$treatment
ind_system <- ind_meta$farming_system

length(unique(ind_treatment))
length(unique(ind_time))
length(unique(ind_system))

#A = 1.0000: Species is a good indicator for the Group, because it occurs ONLY in sites, belonging to this group
#B: proportion of sites belonging to the group that contain the species.

#indicators for farming system 

indval_System = multipatt(ind_analysis,ind_system,func = "IndVal", control = how(nperm=999))
summary(indval_System,indvalcomp=TRUE)
summary(indval_System, alpha=1,indvalcomp=TRUE)

#indicators for drought treatment

indval_Treatment = multipatt(ind_analysis,ind_treatment,func = "IndVal", control = how(nperm=999))
summary(indval_Treatment,indvalcomp=TRUE)
summary(indval_Treatment, alpha=1,indvalcomp=TRUE)

#indicators for the time of sampling

indval_Time = multipatt(ind_analysis,ind_time,func = "IndVal", control = how(nperm=999))
summary(indval_Time,indvalcomp=TRUE)
summary(indval_Time, alpha=1,indvalcomp=TRUE)

```
**4 indicators for farming system, only for CONMIN, 3 indicators for the drought treatment (R=1, C=2), 2 indicators for time, only T1.**


## indicators for the farming systems

```{r indicators system, echo=F}

indicators_system <- indval_System$sign

ind_BIODYN <- as.matrix(indicators_system[which(indicators_system$s.BIODYN == 1 & indicators_system$p.value < 0.05),])
ind_CONMIN <- as.matrix(indicators_system[which(indicators_system$s.CONMIN == 1 & indicators_system$p.value < 0.05),])

CONMIN_BIODYN <- rbind(ind_BIODYN, ind_CONMIN)
colnames(CONMIN_BIODYN)[1:2] <-c("BIODYN","CONMIN")

## Total number of indicator ASVs
length(unique(rownames(CONMIN_BIODYN)))

```
**4 indicators for CONMIN.**

## indicators for the drought treatment

```{r indicators treatment, echo=F}

indicators_treatment <- indval_Treatment$sign

ind_R <- as.matrix(indicators_treatment[which(indicators_treatment$s.R == 1 & indicators_treatment$p.value < 0.05),])
ind_C <- as.matrix(indicators_treatment[which(indicators_treatment$s.C == 1 & indicators_treatment$p.value < 0.05),])
ind_RC <- as.matrix(indicators_treatment[which(indicators_treatment$s.RC == 1 & indicators_treatment$p.value < 0.05),])

R_C_RC <- rbind(ind_R, ind_C, ind_RC)
colnames(R_C_RC)[1:3] <-c("C","R","RC")

## Total number of indicator OTUS
length(unique(rownames(R_C_RC)))

```
**3 indicators for the drought treatment, 2 for C and 1 for R.**

## indicators for the time of sampling

```{r time, echo=F}

indicators_time <- indval_Time$sign

ind_T1 <- as.matrix(indicators_time[which(indicators_time$s.T1 == 1 & indicators_time$p.value < 0.05),])
ind_T3 <- as.matrix(indicators_time[which(indicators_time$s.T3 == 1 & indicators_time$p.value < 0.05),])

T1_T3 <- rbind(ind_T1, ind_T3)
colnames(T1_T3)[1:2] <-c("T1", "T3")

## Total number of indicator OTUS
length(unique(rownames(T1_T3)))

```
**2 indicators for T1.**

# TAXONOMIC COMPOSITION on different taxonomic levels

```{r}

View(unite_ASV_rel@otu_table)

```

## Class level

```{r, echo=F}

library(ggpubr)

#taxonomy_AMF
#we remove unassigned, in particular for stats

unique(tax_table(unite_ASV_rel)[,3])

subset_class <- subset_taxa(unite_ASV_rel, Class %in% c("c__Paraglomeromycetes", "c__Glomeromycetes", "c__Archaeosporomycetes")) 

subset_class
subset_class = prune_samples(sample_sums(subset_class)>0, subset_class)
data_class <- subset_class %>%
              tax_glom(taxrank = "Class") %>%     #agglomerate at class level
              psmelt() %>%                        #melt to long format
              filter(Abundance > 0.01) %>%        #small group (<1%) are removed
              arrange(Class)                      #sort data frame alphabetically by class

#PERMANOVA
set.seed(8899)
AMF_class_dist <- phyloseq::distance(subset_class, method="bray")
adonis_AMF_class <- adonis2(AMF_class_dist ~ treatment+time+farming_system+
                                             farming_system:treatment+treatment:time,
                                             time:farming_system,                                                                                                 strata=meta.ASV$block,
                                             data=meta.ASV,permutations = 999)
adonis_AMF_class
#time: 0.033

adonis_AMF_class.df <- as.data.frame(adonis_AMF_class)
adonis_AMF_class.df<-adonis_AMF_class.df[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(adonis_AMF_class.df,"adonis_AMF_class.txt",sep="\t",quote=F)

#plot AMF class 

boxplot_class_treatment <- ggplot(data_class, aes(x=Class, y=Abundance, fill=treatment)) + stat_boxplot(geom = "errorbar") + xlab("Class") + ylab("Relative abundance") + scale_x_discrete(limits=c("c__Glomeromycetes","c__Paraglomeromycetes","c__Archaeosporomycetes")) + geom_boxplot() + scale_fill_brewer(palette = "Greys") + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Drought treatment") + scale_y_continuous(limits=c(0.00,1.00))

print(boxplot_class_treatment)
ggsave("class_drought_final.tiff", width = 18, height = 10, units = "cm")

boxplot_class_system <- ggplot(data_class, aes(x=Class, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Class") + ylab("Relative abundance") + scale_x_discrete(limits=c("c__Glomeromycetes","c__Paraglomeromycetes","c__Archaeosporomycetes")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Farming system") + scale_y_continuous(limits=c(0.00,1.00))

print(boxplot_class_system)
ggsave("class_system_final.tiff", width = 18, height = 10, units = "cm")

#final figure for the publication

boxplot_class_treat_sys <- ggplot(data_class, aes(x=Class, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Class") + ylab("Relative abundance")+ scale_x_discrete(limits=c("c__Glomeromycetes","c__Paraglomeromycetes", "c__Archaeosporomycetes")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill = "Farming system") + facet_wrap(.~treatment, scales="free_x") + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(limits=c(0.00,1.00))

print(boxplot_class_treat_sys)

ggsave("class_level_final.pdf", width = 40, height = 20, units = "cm")
ggsave("class_level_final.tiff", width = 18, height = 10, units = "cm")

```
**Three AMF classes were extracted. Taxonomic composition on AMF class level is significantly affected by the time of sampling. Boxplots show the relative abundance of AMF classes in the two farming systems under the drought treatments.**

## Focus on Glomeromycetes class and three orders within it 

```{r, echo=FALSE}

unique(tax_table(unite_ASV_rel)[,4])

subset_orders <- subset_taxa(unite_ASV_rel, Order %in% c("o__Diversisporales","o__Glomerales","o__Gigasporales"))

glomeromycetes_orders = prune_samples(sample_sums(subset_orders)>0, subset_orders)

glomeromycetes_orders <- glomeromycetes_orders %>%
                         tax_glom(taxrank = "Order") %>%      #agglomerate at order level
                         psmelt() %>%                         #melt to long format
                         filter(Abundance > 0.01) %>%         #small group (<1%) are removed
                         arrange(Order)                       #sort data frame alphabetically by order

#final figure for the publication

glomeromycetes_plot <- ggplot(glomeromycetes_orders, aes(x=Order, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("AMF orders within Glomeromycetes") + ylab("Relative abundance") +  scale_x_discrete(limits=c("o__Diversisporales","o__Glomerales")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(.~treatment, scales="free_x") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme(axis.text.x = element_text(size = 10)) + scale_y_continuous(limits=c(0.00,1.00)) + labs(fill="Farming system")

print(glomeromycetes_plot)

ggsave("glomeromycetes_only.pdf", width = 40, height = 20, units = "cm")
ggsave("glomeromycetes_only.tiff", width = 18, height = 10, units = "cm")

boxplot_glomeromycetes_treatment <- ggplot(glomeromycetes_orders, aes(x=Order, y=Abundance, fill=treatment)) + stat_boxplot(geom = "errorbar") + xlab("AMF orders within Glomeromycetes") + ylab("Relative abundance") + scale_x_discrete(limits=c("o__Diversisporales","o__Glomerales")) + geom_boxplot() + scale_fill_brewer(palette = "Greys") + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Drought treatment") + scale_y_continuous(limits=c(0.00,1.00))

print(boxplot_glomeromycetes_treatment)
ggsave("glomeromycetes_only_treat.tiff", width = 18, height = 10, units = "cm")

#boxplot system

boxplot_glomeromycetes_system <- ggplot(glomeromycetes_orders, aes(x=Order, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("AMF orders within Glomeromycetes") + ylab("Relative abundance") + scale_x_discrete(limits=c("o__Diversisporales","o__Glomerales")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Farming system") + scale_y_continuous(limits=c(0.00,1.00)) + theme(axis.text.x = element_text(angle = 35, hjust = 1))

print(boxplot_glomeromycetes_system)
ggsave("glomeromycetes_only_system.tiff", width = 18, height = 10, units = "cm")

#PERMANOVA
set.seed(1556)
AMF_glomeromycetes_dist <-phyloseq::distance(subset_orders, method="bray")
adonis_glomeromycetes <-adonis2(AMF_glomeromycetes_dist ~ 
                                treatment+time+farming_system+
                                farming_system:treatment+treatment:time,time:farming_system,                                                         strata=meta.ASV$block,
                                data=meta.ASV,permutations = 999)
adonis_glomeromycetes
#no effect

adonis_glomeromycetes.df <- as.data.frame(adonis_glomeromycetes)
adonis_glomeromycetes.df<-adonis_glomeromycetes.df[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(adonis_glomeromycetes.df,"adonis_glomeromycetes.txt",sep="\t",quote=F)

```
**Three orders within class Glomeromycetes were investigated. They were significantly affected by the time of sampling. Boxplot show the relative abundance of three orders, interestingly Gigasporales under RC was only in BIODYN.**

## Family level 

```{r, echo=F}

unique(tax_table(unite_ASV_rel)[,5])

subset_family <- subset_taxa(unite_ASV_rel, Family %in% c("f__Archaeosporaceae", "f__Glomeraceae", "f__Diversisporaceae", "f__Ambisporaceae", "f__Acaulosporaceae", "f__Paraglomeraceae", "f__Gigasporaceae", "f__Claroideoglomeraceae")) 

subset_family
subset_family = prune_samples(sample_sums(subset_family)>0, subset_family)

data_family <- subset_family %>%
               tax_glom(taxrank = "Family") %>%
               psmelt() %>%
               filter(Abundance > 0.01) %>%
               arrange(Family)

#PERMANOVA
set.seed(8899)
AMF_family_dist <-phyloseq::distance(subset_family, method="bray")
adonis_AMF_family <-adonis2(AMF_family_dist ~ treatment+time+farming_system+
                                            farming_system:treatment+treatment:time,
                                            time:farming_system,                                                                                                 strata=meta.ASV$block,
                                            data=meta.ASV,permutations = 999)
adonis_AMF_family
#time: 0.022

adonis_AMF_family.df <- as.data.frame(adonis_AMF_family)
adonis_AMF_family.df<-adonis_AMF_family.df[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(adonis_AMF_family.df,"adonis_AMF_family.txt",sep="\t",quote=F)

#boxplot treatment

boxplot_family_treatment <- ggplot(data_family, aes(x=Family, y=Abundance, fill=treatment)) + stat_boxplot(geom = "errorbar") + xlab("Family") + ylab("Relative abundance") + scale_x_discrete(limits=c("f__Acaulosporaceae", "f__Glomeraceae", "f__Diversisporaceae", "f__Archaeosporaceae", "f__Ambisporaceae", "f__Claroideoglomeraceae", "f__Paraglomeraceae")) + geom_boxplot() + scale_fill_brewer(palette = "Greys") + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Drought treatment") + scale_y_continuous(limits=c(0.00,1.00)) + theme(axis.text.x = element_text(angle = 35, hjust = 1))

print(boxplot_family_treatment)
ggsave("family_level_treat.tiff", width = 18, height = 10, units = "cm")

#boxplot system

boxplot_family_system <- ggplot(data_family, aes(x=Family, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Family") + ylab("Relative abundance") + scale_x_discrete(limits=c("f__Acaulosporaceae", "f__Glomeraceae", "f__Diversisporaceae", "f__Archaeosporaceae", "f__Ambisporaceae", "f__Claroideoglomeraceae", "f__Paraglomeraceae")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Farming system") + scale_y_continuous(limits=c(0.00,1.00)) + theme(axis.text.x = element_text(angle = 35, hjust = 1))

print(boxplot_family_system)
ggsave("family_level_system.tiff", width = 18, height = 10, units = "cm")

#boxplot merged

boxplot_family_treat_sys <- ggplot(data_family, aes(x=Family, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Family") + ylab("Relative abundance")+ scale_x_discrete(limits=c("f__Acaulosporaceae", "f__Glomeraceae", "f__Diversisporaceae", "f__Archaeosporaceae", "f__Ambisporaceae", "f__Claroideoglomeraceae", "f__Paraglomeraceae")) + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + theme(axis.text.x = element_text(size = 7)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
theme(axis.text.x = element_text(angle = 45, hjust = 1)) + facet_wrap(.~treatment, scales="free_x") + theme(axis.text.x = element_text(size = 10)) + labs(fill="Farming system") + scale_y_continuous(limits=c(0.00,1.00))

print(boxplot_family_treat_sys)
ggsave("family_level_final.tiff", width = 20, height = 10, units = "cm")

```
**When investigating all 8 AMF families, they are significantly affected by the time of sampling.**

## Genus level 

```{r, echo=F}

#here we will include unassigned

unique(tax_table(unite_ASV_rel)[,6])

subset_genera <- subset_taxa(unite_ASV_rel, Genus %in% c("g__Acaulospora", "g__Glomus", "g__Palaeospora", "NA", "g__Diversispora", "g__Funneliformis", "g__Paraglomus", "g__Ambispora", "g__Claroideoglomus", "g__Gigaspora", "Cetraspora", "g__Dominikia", "g__Archaeospora"))

subset_genera = prune_samples(sample_sums(subset_genera)>0, subset_genera)
data_genera <- subset_genera %>%
               tax_glom(taxrank = "Genus") %>%
               psmelt() %>%
               filter(Abundance > 0.01) %>%
               arrange (Genus)
               
#PERMANOVA
set.seed(8899)
AMF_genus_dist <-phyloseq::distance(subset_genera, method="bray")
adonis_AMF_genus <-adonis2(AMF_genus_dist ~ treatment+time+farming_system+
                                            farming_system:treatment+treatment:time,
                                            time:farming_system,                                                                                                 strata=meta.ASV$block,
                                            data=meta.ASV,permutations = 999)
adonis_AMF_genus
#only time effect

adonis_AMF_genus.df <- as.data.frame(adonis_AMF_genus)
adonis_AMF_genus.df<-adonis_AMF_genus.df[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(adonis_AMF_genus.df,"adonis_AMF_genus.txt",sep="\t",quote=F)

#boxplot

genera_plot <- ggplot(data_genera, aes(x=Genus, y=Abundance, fill=farming_system)) + stat_boxplot(geom = "errorbar") + xlab("Genus") + ylab("Relative abundance") + geom_boxplot() + scale_fill_manual(values=c("grey70", "grey25")) + scale_x_discrete(limits=c("g__Acaulospora", "g__Paraglomus", "g__Diversispora", "g__Funneliformis", "g__Ambispora", "g__Palaeospora", "g__Claroideoglomus", "g__Archaeospora")) + theme(axis.text.x = element_text(size = 8)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + labs(fill="Farming system") + scale_y_continuous(limits=c(0.00,1.00))

ggsave("genus_level_final.pdf", width = 40, height = 20, units = "cm")
ggsave("genus_level_final.tiff", width = 18, height = 10, units = "cm")

#additional 
genera_plot + facet_wrap(.~treatment, scales="free_x") + theme(axis.text.x = element_text(size = 10))

genera_treatment <- ggplot(data_genera, aes(x=Genus, y=Abundance, fill=treatment)) + stat_boxplot(geom = "errorbar") + xlab("Genus") + ylab("Relative abundance") + scale_x_discrete(limits=c("g__Acaulospora", "g__Paraglomus", "g__Diversispora", "g__Funneliformis", "g__Ambispora", "g__Palaeospora", "g__Claroideoglomus")) + geom_boxplot() + scale_fill_brewer(palette = "Greys") + theme(axis.text.x = element_text(size = 7.5)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + labs(fill="Drought treatment") + scale_y_continuous(limits=c(0.00,1.00)) + theme(axis.text.x = element_text(angle = 35, hjust = 1))

print(genera_treatment)
ggsave("genera_treat.tiff", width = 18, height = 10, units = "cm")

```
**When filtering on the abundance is performed, we kept 9 genera (8 assigned + 1 unassigned). Only signficant effect of the time of sampling.**

# Session Info

```{r sessionInfo}
sessionInfo()
```

