---
title: "AMF communities in DOK (SoilClim)"
author: "Katja Kozjek"
date: "`r Sys.Date()`"
output: 
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(warning = FALSE, message = FALSE)

```

# Prepare the environment

```{r setting the environment, message=F, warning=F, echo=F}

#load libraries
library(dada2); packageVersion("dada2") 
library(phyloseq); packageVersion("phyloseq") 
library(vegan); packageVersion("vegan") 
library(ggplot2); packageVersion("ggplot2")
library(dplyr); packageVersion("dplyr")
library(tidyverse); packageVersion("tidyverse")
library(ggpubr); packageVersion("ggpubr")
library(devtools); packageVersion("devtools")
library(ape); packageVersion("ape")
library(brms); packageVersion("brms") 
library(shinystan); packageVersion("shinystan") 
library(picante); packageVersion("picante")
library(indicspecies); packageVersion("indicspecies")
library(ggvegan); packageVersion("ggvegan")
library(devtools); packageVersion("devtools")
library(ggrepel); packageVersion("ggrepel")
library(ranacapa); packageVersion("ranacapa")

```

# Start with DADA2 pipeline, create ASVs from fastq files
## set path

```{r set path, echo=F}

path_fastq <- "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/samples"
list.files(path_fastq) #listing all files in the path
fns <- sort(list.files(path_fastq, pattern=".fq", full.names = TRUE)) #selection of all fastq files
sample.names <- sapply(strsplit(basename(fns), "_"), `[`, 2)
print(sample.names)

```

## visualize sequence quality 

```{r sequence quality, echo=F}

plotQualityProfile(fns[1:6])
plotQualityProfile(fns[7:12])
plotQualityProfile(fns[13:18])
plotQualityProfile(fns[19:24])
plotQualityProfile(fns[25:30])
plotQualityProfile(fns[31:36])
plotQualityProfile(fns[37:42])
plotQualityProfile(fns[43:48])

```

## length distribution of the original sequences 

```{r original length distribution, echo=F}

#inspect length distribution of sequences to determine how to filter the sequences
length_fns <- lapply(path_fastq, function(fn) nchar(getSequences(fn)))
length <- do.call(c, length_fns)
hist(length, 100) #looks good, majority between 1400 and 1700
mean(length); median(length) 

```

## filter and trim original sequences 

```{r filter and trim, echo=F}

#place filtered files in filtered/ subdirectory 
filt_path <- file.path(path_fastq, "filtered") 
filt_fns <- file.path(path_fastq, "filtered", paste0(sample.names, "_filt.fq.gz"))
names(filt_fns) <- sample.names

#filter from 1400 to 1650
filt_out <- filterAndTrim(fns, filt_fns, minQ=2, minLen=1450, maxLen=1650, maxN=0, rm.phix=FALSE, maxEE=3, multithread = FALSE) #we retain only reads with fewer than three exp. errors 
print(filt_out)

#csv file containing number of input and output/filtered sequences
write.csv(filt_out, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results/filtered_seq.csv") 

```

## inspect length distribution of filtered sequences

```{r length distribution filtered sequences, echo=F}

length_fns_filt <- lapply(filt_path, function(fn) nchar(getSequences(fn)))
length_filt <- do.call(c, length_fns_filt)
hist(length_filt, 100)
mean(length_filt); median(length_filt)
#1529.076
#1529

```

## dereplication of identical reads
 + dereplication combines all identical sequencing reads into into “unique sequences” with a corresponding “abundance”: the number of reads with that unique sequence
+ collapsing together all reads that encode the same sequence

```{r dereplication, echo=F}

#collapse identical sequences
derep <- derepFastq(filt_fns, verbose=TRUE)
names(derep) <- sample.names #name the derep-class objects by the sample names

```

## error rates
+ error rates for each possible transition

```{r error rates, echo=F}

set.seed(100)
err <- learnErrors(derep, errorEstimationFunction = PacBioErrfun, BAND_SIZE = 32, verbose = TRUE)  
#103155500 total bases in 67095 reads from 12 samples will be used for learning the error rates.

plotErrors(err)
plotErrors(err, nominalQ=TRUE) #red line shows the error rates, the estimated error rates (black line) are a good fit to the observed rates (points), error rates should drop with the increased quality 

```

## denoise 
+ infer the sequence variants in each sample
+ separate true biological sequences from errors

```{r denoise, echo=F}

dada_dd <- dada(derep, err = err, BAND_SIZE = 32, multithread=TRUE, verbose = TRUE, pool = "pseudo") #number of true sequences
dada_dd[[1]] #check only the first sample

```

# Construct sequence variant (ASV) table, a higher-resolution version of the OTU table produced by traditional methods

```{r ASV table, echo=F}

seqtab <- makeSequenceTable(dada_dd); dim(seqtab) #48 972
table(nchar(getSequences(seqtab))) #distibution of sequence lengths 
#shortest 1468, longest 1634

#save R object containing all original sequences
saveRDS(seqtab, "data/rds/original_seq.rds")

#extract original sequences, belonging to 972 ASVs

asv_seqs_original <- colnames(seqtab)
asv_headers_original <- vector(dim(seqtab)[2], mode = "character")
for (i in 1:dim(seqtab)[2]) {
  asv_headers_original[i] <- paste(">ASV", i, sep = "_")
}
asv_fasta_original <- c(rbind(asv_headers_original, asv_seqs_original))
write(asv_fasta_original, "data/fasta/original_seq.fasta")

```

## remove chimeras 

```{r remove chimeras, echo=F}

seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim) 
#48 962
#chimeras removed 
#seqtab.nochim file is the final file, samples and sequences that represents ASVs

saveRDS(seqtab.nochim, "data/rds/nochim_seq.rds")

#what proportion of the sequence variants are chimeras
sum(seqtab.nochim)/sum(seqtab) 
#0.9958734
1-sum(seqtab.nochim)/sum(seqtab)  #0.004126584

```

## summary, track retained sequences

```{r summarize, echo=F}

getN <- function(x) sum(getUniques(x))
track <- cbind(filt_out, sapply(dada_dd, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoised", "nonchim")
rownames(track) <- sample.names
print(track)
class(track) #it is a matrix

#save summary 
write.csv(track, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results/summary_seq.csv")

#plot the distribution of sequence length 
plot(sort(unname(rowSums(seqtab.nochim))))

#species richness
rarecurve(seqtab.nochim)

```

# Export fasta file 

```{r fasta files, echo=F}

#give our sequence headers more manageable names (ASV_1, ASV_2...) and write out a FASTA of our ASV seqs
asv_seqs <- colnames(seqtab.nochim)
asv_headers <- vector(dim(seqtab.nochim)[2], mode = "character")
for (i in 1:dim(seqtab.nochim)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep = "_")
}
asv_fasta <- c(rbind(asv_headers, asv_seqs))

#export fasta file with all 962 ASVs

write(asv_fasta, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/data/fasta/processed_seq.fasta")

```

# Export count table

```{r count table, echo=F}

asv_tab <- (seqtab.nochim)
colnames(asv_tab) <- sub(">", "", asv_headers)

write.table(asv_tab, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/data/csv/ASVs_counts.csv", sep = ",", quote = FALSE, col.names = NA)

#matrix: rows corresponding to samples, columns with ASV, the value of each entry is the number of times that ASV was observed in the sample

```

# Taxonomy with UNITE 2020

```{r taxonomy unite 2020, echo=F}

set.seed(NULL) #return to the original state by unsetting the seed
set.seed(62)

#take UNITE database

taxonomy_unite <- assignTaxonomy(seqtab.nochim, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/reference databases/sh_general_release_dynamic_04.02.2020_dev.fasta", multithread=TRUE, tryRC = TRUE)

tax_unite2020 <- taxonomy_unite # Removing sequence rownames for display only
rownames(tax_unite2020) <- NULL
rownames(tax_unite2020) <- getSequences(asv_tab)
head(tax_unite2020)

write.csv(tax_unite2020, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/data/csv/tax_unite2020.csv")

write.csv(taxonomy_unite, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/data/csv/tax_unite2020_seq.csv")

```

# Create phyloseq object with 962 ASV, based on UNITE taxonomy

```{r create phyloseq, echo=F}

#OTU table
OTU = otu_table(asv_tab, taxa_are_rows = FALSE)
#962 taxa, 48 samples
OTUr = transform_sample_counts(OTU, function(x) x/sum(x))

#TAX table from UNITE

taxa_unite <- read.csv("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/data/csv/tax_unite2020.csv", sep = ",", row.names = 1)

tax_unite_table = tax_table(as.matrix(taxa_unite))
TAX = tax_table(tax_unite_table)

#META file 
meta_file = read.csv("/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/Meta files AMF/meta_DOK.csv", sep=",", row.names=1) 
meta_file = sample_data(data.frame(meta_file))

#create phyloseq

physeq_unite = phyloseq(OTU, TAX, meta_file)
physeq_unite
#OTU_table; 962 taxa, 48 samples
#tax_table; 962 taxa, by 7 taxonomic ranks
#meta_file; 48 samples, 24 variables 

#phyloseq with relative abundance
physeq_unite_rel = phyloseq(OTUr, TAX, meta_file)

#save unite phyloseq object with relative abundance
saveRDS(physeq_unite_rel, "data/rds/phyloseq_unite_rel.rds")

sort(sample_sums(physeq_unite), decreasing = F) 
sum(otu_table(physeq_unite)) #185825 sequencing reads in all samples

#add tree
unite_tree = rtree(ntaxa(physeq_unite), rooted=TRUE, tip.label=taxa_names(physeq_unite))
plot((unite_tree))

#add unite tree to phyloseq object 
physeq_unite = merge_phyloseq(physeq_unite, unite_tree)
physeq_unite

#save unite phyloseq object
saveRDS(physeq_unite, "data/rds/phyloseq_unite.rds")

#plot relative abundance 
plot_bar(tax_glom(physeq_unite_rel, "Phylum"), fill="Phylum")
plot_bar(tax_glom(physeq_unite_rel, "Phylum"), fill="Phylum") + facet_grid(.~treatment *time, scale="free_x", space = "free_x")

```
**Initial phyloseq object with 962 ASVs in 48 samples is created. Total number of sequences is 185,825.**

## filter phyloseq object

```{r filter phyloseq object, echo=F}

#remove non-AMF 
physeq_unite_Filtered <- subset_taxa(physeq_unite, !Phylum %in% c ("p__Basidiomycota", "p__Ascomycota")) #we remove non-AMF and now we have 955 ASVs
physeq_unite_Filtered

physeq_unite_Filtered_rel <- subset_taxa(physeq_unite_rel, !Phylum %in% c ("p__Basidiomycota", "p__Ascomycota"))

plot_bar(tax_glom(physeq_unite_Filtered_rel, "Phylum"), fill="Phylum")

sum(sample_sums(physeq_unite)) #185825
sum(sample_sums(physeq_unite_Filtered)) #185364
sort(sample_sums(physeq_unite_Filtered), decreasing = F) #we will discard samples with less than 500 reads 

```

## phyloseq object with only AMF (Glomeromycota)

```{r AMF phyloseq object, echo=F}

#create new phyloseq 
physeq_AMF <- physeq_unite_Filtered

#save phyloseq object
saveRDS(physeq_AMF, "data/rds/phyloseq_AMF.rds")

write.csv(physeq_AMF@otu_table, "data/csv/AMF_count.csv")
write.csv(physeq_AMF@tax_table, "data/csv/tax_AMF.csv")

#compute prevalence of each feature, store as data.frame
prevdf=apply(X=otu_table(physeq_AMF), 
             MARGIN=ifelse(taxa_are_rows(physeq_AMF), yes=1, no=2),
             FUN=function(x){sum(x>0)})

#add taxonomy and total read frames to this data.frame
prevdf=data.frame(Prevalence=prevdf,
                  TotalAbundance=taxa_sums(physeq_AMF), tax_table(physeq_AMF))


write.csv(prevdf,"results/prevalance_rel.csv")

#relative prevalence
#compute prevalence of each feature, store as data.frame
prevdf_rel=apply(X=otu_table(physeq_unite_Filtered_rel), 
             MARGIN=ifelse(taxa_are_rows(physeq_unite_Filtered_rel), yes=1, no=2),
             FUN=function(x){sum(x>0)})

#add taxonomy and total read frames to this data.frame
prevdf_rel=data.frame(Prevalence=prevdf_rel,
                  TotalAbundance=taxa_sums(physeq_unite_Filtered_rel), tax_table(physeq_unite_Filtered_rel))

write.csv(prevdf_rel,"results/prevalance_rel.csv")

```

## summarize 

```{r summary of amf dataset, echo=F}

min(sample_sums(physeq_AMF)) #404
max(sample_sums(physeq_AMF)) #11426
median(sample_sums(physeq_AMF)) #3414.5
mean(sample_sums(physeq_AMF)) #3861.75

#how many taxa are there? 
unique(ntaxa(physeq_AMF)) #955

```
**955 ASVs in Glomeromycota phylum**

## rarefaction curves

```{r rarefaction curves, echo=F}

barplot(sort(sample_sums(physeq_AMF)),
        ylab= "Number of reads", xlab="Sample ID")

rare_curve <- ggrare(physeq_AMF, step = 10, se = TRUE)

rare_curve <- rare_curve + xlab("Sequence sample size") + ylab("Number of ASVs") + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) 
print(rare_curve)

ggsave("results/rarefaction_curves.tiff", width = 15, height = 10, units = "cm")
ggsave("results/rarefaction_curves.jpeg", width = 15, height = 10, units = "cm")
ggsave("results/rarefaction_curves.pdf", width = 15, height = 10, units = "cm")

```
**To determine the completeness of our sampling efforts, we plotted rarefaction curves for the 48 samples. Sufficient sequencing depth to recover the majority of AMF ASVs reached.**

#Convert from phyloseq to vegan or from vegan to phyloseq

```{r vegan_phyloseq functions, echo=F}

# convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

```

# Prepare ASV level 
+ remove samples with less than 500 sequences
+ TSS normalization 

```{r asv level, echo=F}

sort(sample_sums(physeq_AMF), decreasing = F)

physeq_unite_Filtered_ASV <- prune_samples(sample_sums(physeq_unite_Filtered) >=500, physeq_AMF)

plot_bar((physeq_unite_Filtered_ASV))
unique(ntaxa(physeq_unite_Filtered_ASV)) 

#TSS normalization
unite_ASV_rel <- transform_sample_counts(physeq_unite_Filtered_ASV, function(x) x/sum(x))

```

# COMMUNITY COMPOSITION 
+ here I explore the community composition on genus level
+ agglomerate on genus level
+ PERMANOVA with Bray-Curtis
+ plot PCoA/NMDS

## GENUS level 
+ agglomerate on genus level 
+ remove samples with less than 500 reads
+ TSS normalization

```{r community composition genus level, echo=F}

## agglomerate on genus level 

length(get_taxa_unique(physeq_AMF, taxonomic.rank = "Genus"))

# agglomerate taxa at the genus level (combine all with the same name) and remove all taxa without genus level assignment

physeq_AMF_genus = tax_glom(physeq_AMF, "Genus", NArm = T)
physeq_AMF_genus #agglomerated on the genus level

sum(sample_sums(physeq_AMF_genus)) #128352

unique(tax_table(physeq_AMF_genus)[,6]) #12

sort(sample_sums(physeq_AMF_genus), decreasing = F) #we will remove 3 samples

#keep samples with more than 500 reads 
physeq_AMF_genus_500 <- prune_samples(sample_sums(physeq_AMF_genus) >=500, physeq_AMF_genus)

sort(sample_sums(physeq_AMF_genus_500))
plot_bar((physeq_AMF_genus_500))
unique(ntaxa(physeq_AMF_genus_500)) #12
any(taxa_sums(physeq_AMF_genus_500) == 0)

#TSS normalization 
physeq_AMF_genus_500rel <- transform_sample_counts(physeq_AMF_genus_500, function(x) x/sum(x))

``` 

```{r rarefaction, echo=F}

#rarefaction
unite_genus_rare <- rarefy_even_depth(physeq_AMF_genus_500, sample.size = min(sample_sums(physeq_AMF_genus_500)), 6020, replace=F, verbose = T) # rarefication on the lowest number of reads 

sort(sample_sums(unite_genus_rare))

```

# NMDS ordination (genus level)

```{r nmds ordination}

#transfer OTU table from phyloseq into a vegan object
vegan_AMF_genus<-psotu2veg(physeq_AMF_genus_500rel)

#transfer sample data from phyloseq into a vegan object
veganSample_AMF_genus<-pssd2veg(physeq_AMF_genus_500rel)

#rename columns from ASV to genus name 
colnames(vegan_AMF_genus) <- c("g__Paraglomus",  "g__Acaulospora", "g__Septoglomus", "g__Glomus", "g__Claroideoglomus", "g__Gigaspora", "g__Funneliformis", "g__Diversispora", "g__Archaeospora", "g__Ambispora", "g__Dominikia",  "g__Palaeospora")
vegan_AMF_genus

#calculate the distance matrix
set.seed(0902)
nmds_ord <- metaMDS(vegan_AMF_genus, distance = "bray", k=2, trymax=1000, autotransform=T, permutations = 999)
nmds_ord$stress
nmds_ord$species

plot_ordination(physeq_AMF_genus_500rel, nmds_ord, color = "farming_system", shape = "time") + geom_point(size = 3) + theme_bw()

#stress is goodness of fit 
goodness(nmds_ord)
stressplot(nmds_ord)
 
#this function draws NMDS ordination diagram with sites (=my samples)
plot(nmds_ord)
plot(nmds_ord, display = 'sites', type = 't')
plot(nmds_ord, display = 'species', type = 't')

#species spores 
envfit(nmds_ord, vegan_AMF_genus) 

plot(nmds_ord, type='n')
points(nmds_ord, display=c('sites'), choices=c(1, 2), pch=3, col='red')
text(nmds_ord, display=c('species'), choices=c(1, 2), col='blue', cex=0.7)

```

```{r, echo=F}
#add volumetric soil water content (VWC)
veganSample_AMF_genus$VWC<-veganSample_AMF_genus$WC*veganSample_AMF_genus$BD
summary(veganSample_AMF_genus$VWC)

#rename
names(veganSample_AMF_genus)[names(veganSample_AMF_genus) == "SOC"] <- "total.C"

```

## prepare for envfit 

```{r calculate envfit, echo=F}

#envfit with metadata 
nmds_meta <- as(sample_data(physeq_AMF_genus_500rel), "data.frame")
env_nmds <- nmds_meta[,5:24]

data.scores_nmds = as.data.frame(scores(nmds_ord, display = 'sites')) #save NMDS to dataframe
species.scores_nmds = as.data.frame(scores(nmds_ord, display = 'species'))

#add grouping variable to dataframe
data.scores_nmds <- cbind(as.data.frame(data.scores_nmds), farming_system = env_nmds$farming_system, 
                          treatment = env_nmds$treatment)
head(data.scores_nmds)

```

## calculate envfit for ndms ordination
+ investigate the soil parameters which may be driving the distribution pattern

```{r envfit for nmds, echo=F}

#calculate envfit
set.seed(0935)
env.fit <- envfit(scores(nmds_ord), env_nmds, permutations = 999, na.rm = TRUE, strata=nmds_meta$block)
env.fit

#extract scores from environment
env.scores <- as.data.frame(scores(env.fit, display = "vectors"))
env.scores <- cbind(env.scores, env.variables = rownames(env.scores), pval = env.fit$vectors$pvals)
head(env.scores)
sig.env <- subset(env.scores, pval<=0.05) #extract only significant soil parameters
head(sig.env)

#envfit write table for paper

envfit_nmds <- data.frame((env.fit$vectors)$r, (env.fit$vectors)$pvals)
write.table(envfit_nmds, "/Users/Katja/Box Sync/PhD Lund Katja/Sequencing/PacBio AMF/Sequencing data AMF/dada2/AMF_DOK trial/PacBio_AMF_workflow/results/envfit_nmds.txt", sep = "\t", quote = FALSE, row.names = T, col.names=T)

#vectors as continuous, factors as categorical
#vectors are soil parameters and factors (centroids) are treatment + farming system
en_coord_cont = as.data.frame(scores(env.fit, "vectors")) * ordiArrowMul(env.fit)
en_coord_cat = as.data.frame(scores(env.fit, "factors")) * ordiArrowMul(env.fit)

```

## final NMDS plot 

```{r NMDS plot on genus level, echo=F}
#plot with significant environmental parameters

nmds_final <- ggplot() + geom_point(data = data.scores_nmds, aes(x = NMDS1, y = NMDS2, colour = farming_system), size = 5, alpha = 0.5) + theme_bw() + scale_colour_manual("Farming system", values=c("grey70", "grey25")) + theme(axis.title = element_text(size = 12, colour = "black"), axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), legend.title = element_text(size = 11, colour = "black"), legend.text = element_text(size = 10, colour = "black")) + labs(colour = "farming_system") + annotate("text", x=1.0, y=-0.7, size=5,label= paste("Stress =", round(nmds_ord$stress, digits=2))) + geom_segment(data = sig.env, aes(x = 0, xend=NMDS1, y=0, yend=NMDS2), arrow = arrow(length = unit(0.25, "cm")), colour = "black", lwd=0.3) + ggrepel::geom_text_repel(data = sig.env, aes(x=NMDS1, y=NMDS2, label = env.variables), cex = 4, color="black", direction = "both", segment.size = 0.25) + geom_point(data = en_coord_cat, aes(x = NMDS1, y = NMDS2), shape = "diamond", size = 5, alpha = 0.8, colour = "black") + geom_text(data = en_coord_cat, aes(x = NMDS1, y = NMDS2+0.04), label = row.names(en_coord_cat), colour = "black", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

print(nmds_final)
ggsave("results/nmds_final.pdf", width = 40, height = 20, units = "cm")
ggsave("results/nmds_final.tiff", width = 40, height = 20, units = "cm")

#ndms with dots representing genus 

ggplot() + geom_point(data = species.scores_nmds, aes(x = NMDS1, y = NMDS2), size = 5, alpha = 0.5) + theme_bw() + scale_colour_manual("Farming system", values=c("grey70", "grey25")) + theme(axis.title = element_text(size = 12, colour = "black"), axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), legend.title = element_text(size = 11, colour = "black"), legend.text = element_text(size = 10, colour = "black")) + labs(colour = "farming_system") + annotate("text", x=1.0, y=-0.7, size=5,label= paste("Stress =", round(nmds_ord$stress, digits=2))) + geom_segment(data = sig.env, aes(x = 0, xend=NMDS1, y=0, yend=NMDS2), arrow = arrow(length = unit(0.25, "cm")), colour = "black", lwd=0.3) + ggrepel::geom_text_repel(data = sig.env, aes(x=NMDS1, y=NMDS2, label = env.variables), cex = 4, color="black", direction = "both", segment.size = 0.25) + geom_point(data = en_coord_cat, aes(x = NMDS1, y = NMDS2), shape = "diamond", size = 5, alpha = 0.8, colour = "black") + geom_text(data = en_coord_cat, aes(x = NMDS1, y = NMDS2+0.04), label = row.names(en_coord_cat), colour = "black", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

```

# PERMANOVA with BC on genus level 

```{r permanova on genus level, echo=F}

unite_genus_rel <- transform_sample_counts(physeq_AMF_genus_500rel, function(x) x/sum(x))
bray_unite_genus <- phyloseq::distance(unite_genus_rel, method = "bray")

meta.unite_genus <- as(sample_data(physeq_AMF_genus_500rel), "data.frame")

set.seed(1546)
perm.genus <- adonis2(bray_unite_genus ~ treatment+time+farming_system+
                    farming_system:treatment+
                    treatment:time+
                    time:farming_system,
                    strata=meta.unite_genus$block,
                    data = meta.unite_genus,permutations = 999)
perm.genus

#effect of time and farming system on genus composition with bray distance 
#we should highlight with constraint method

#save output for paper
permanova_genus <- as.data.frame(perm.genus)
permanova_genus <- permanova_genus[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(permanova_genus, "results/permanova_genus.txt", sep="\t",quote=F)

```
**Effect of the time and the farming system on genus composition with BC distance.We should highlight these effects with constrained method.**

# weighted Unifrac distance on genus level
+ Weighted UniFrac measure is based on the abundance of different phylogenetic groups

```{r weighted unifrac distance genus, echo=F}

wuni_genus <- UniFrac(physeq_AMF_genus_500, weighted = TRUE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.wuni_genus <- as(sample_data(physeq_AMF_genus_500), "data.frame")

set.seed(1420)
perm_mod_wuni_genus <- adonis2(wuni_genus ~ treatment+time+farming_system+
                                       farming_system:treatment+
                                       treatment:time+
                                       time:farming_system,
                                       strata=meta.wuni_genus$block,
                                       data = meta.wuni_genus,permutations = 999)
perm_mod_wuni_genus

wuni_genus_results <- as.data.frame(perm_mod_wuni_genus)
wuni_genus_results <- wuni_genus_results[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(wuni_genus_results, "results/weighted_Unifrac.txt", sep="\t",quote=F)

```

# unweighted Unifrac distance on genus level
+ the unweigted measure is based on the presence/absence of the different phylogenetic groups in the soil microbial communities

```{r unweighted unifrac distance genus, echo=F}

uni_genus <- UniFrac(physeq_AMF_genus_500, weighted = FALSE, normalized = TRUE, parallel = FALSE, fast = TRUE)
meta.uni_genus <- as(sample_data(physeq_AMF_genus_500), "data.frame")

set.seed(1420)
perm_mod_uni_genus <- adonis2(uni_genus ~ treatment+time+farming_system+
                                     farming_system:treatment+
                                     treatment:time+
                                     time:farming_system,
                                     strata=meta.uni_genus$block,
                                     data = meta.uni_genus,permutations = 999)
perm_mod_uni_genus

uni_genus_results <- as.data.frame(perm_mod_uni_genus)
uni_genus_results <- uni_genus_results[, c("Df","SumOfSqs", "R2", "F", "Pr(>F)")]
write.table(uni_genus_results, "results/unweighted_Unifrac.txt", sep="\t",quote=F)

```

# multivariate dispersion 
+ As PERMANOVA is sensitive to unequal multivariate dispersion, multivariate group means were tested 

```{r multivariate dispersion, echo=F}

modSystem <- with(veganSample_AMF_genus,  betadisper(bray_unite_genus, farming_system))
modSystem

modDrought <- with(veganSample_AMF_genus,  betadisper(bray_unite_genus, treatment))
modDrought

modTime <- with(veganSample_AMF_genus,  betadisper(bray_unite_genus, time))
modTime

set.seed(1089)
permutest(modSystem)

set.seed(1089)
permutest(modDrought)

set.seed(1089)
permutest(modTime)

```

# Constrained analysis with db-RDA, to highlight effects
+ Plot only the variation in your data that explains changes in community composition

## calculate dbrda

```{r plot dbrda, echo=F}

#create the distance matrix
set.seed(1120)
dbrda <- dbrda(vegan_AMF_genus ~ farming_system+time+Condition(block),veganSample_AMF_genus,dist="bray")
head(scores(dbrda, display = 'sites'))
smry <- summary(dbrda)

sppscores(dbrda) <- vegan_AMF_genus

anova(dbrda)  # overall test of the significant of the analysis
anova(dbrda, by="axis")
anova(dbrda, by="term", permutations = 999)

ordiplot(dbrda,type="t",scaling=2)
ordiplot(dbrda,type="t",scaling=1)

#check constrained and unconstrained variance
dbrda$CCA$tot.chi/dbrda$tot.chi

constrained_eig <- dbrda$CCA$eig/dbrda$tot.chi*100
unconstrained_eig <- dbrda$CA$eig/dbrda$tot.chi*100
expl_var <- c(constrained_eig, unconstrained_eig)
barplot (expl_var[1:20], col = c(rep ('red', length (constrained_eig)), rep ('black', length (unconstrained_eig))),
         las = 2, ylab = '% variation')

#calculate amount of variation explained by the axis
b_xlabel<-paste0("db-RDA1"," ","[", round(((dbrda$CCA$eig[1]/dbrda$CA$tot)*100),1), "%]" )
b_ylabel<-paste0("db-RDA2"," ","[", round(((dbrda$CCA$eig[2]/dbrda$CA$tot)*100),1), "%]" )

#fortify: transfer into data frame that can be used for ggplot
scrs <- fortify(dbrda)

#select site (-> Samples)
sites <- subset(scrs, subset = Score == "sites") 

#select species (-> taxa)
species <- subset(scrs, subset = Score == "species")

(Time<-veganSample_AMF_genus$time) #choose group variable here
(System<-veganSample_AMF_genus$farming_system) #choose shape variable
df <- cbind(sites, Time , System) ## add on the group variable

df$Time <- factor(df$Time, levels=c("4w", "13w"), labels = c("4 weeks", "13 weeks"))

```

## plot dbRDA

```{r dbrda katja, echo=F}

dbrda_katja <- ggplot(df, aes(x = dbRDA1, y = dbRDA2)) + geom_point(aes(colour = System, shape = Time), size = 5,alpha=0.8) +
    scale_color_manual(values = c("dimgrey", "darkgray")) + labs(x = paste0("db-RDA1"," ","[", round(((dbrda$CCA$eig[1]/dbrda$CA$tot)*100),1), "%]" ), y = paste0("db-RDA2"," ","[", round(((dbrda$CCA$eig[2]/dbrda$CA$tot)*100),1), "%]" )) + 
    geom_vline(xintercept = 0, linetype="dotted") +
    geom_hline(yintercept = 0, linetype="dotted") +
    theme_bw() + 
    theme(text = element_text(size = 14)) + 
    theme(axis.title = element_text(size = 12, colour = "black"), axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), legend.title = element_text(size = 11, colour = "black"), legend.text = element_text(size = 10, colour = "black")) + labs(colour = "Farming system", shape= "Sampling time") + xlab("dbRDA1 (18.5% of fitted, 12.4% of total variation)") + ylab("dbRDA2 (7.7% of fitted, 5.3% of total variation)") 
  
print(dbrda_katja)

ggsave("results/dbrda_samples.pdf", width = 20, height = 10, units = "cm", dpi=500)
ggsave("results/dbrda_samples.tiff", width = 20, height = 10, units = "cm", dpi=500)
ggsave("results/dbrda_samples.jpeg", width = 20, height = 10, units = "cm", dpi=500)

```

## add environmental parameters to dbrda

```{r, echo=F}
## environmental variables 

names(veganSample_AMF_genus)
env_dbrda <- veganSample_AMF_genus[,2:25]
env_dbrda <- env_dbrda[,-c(2,3,4,7)]
names(env_dbrda)

#rename environmental parameters
names(env_dbrda)[names(env_dbrda) == "total.P"] <- "total P"
names(env_dbrda)[names(env_dbrda) == "total.N"] <- "total N"
names(env_dbrda)[names(env_dbrda) == "total.C"] <- "total C"
names(env_dbrda)[names(env_dbrda) == "total.C"] <- "total C"
names(env_dbrda)[names(env_dbrda) == "C.N"] <- "C:N ratio"
names(env_dbrda)[names(env_dbrda) == "NLFA"] <- "AMF biomass"
names(env_dbrda)[names(env_dbrda) == "N.roots"] <- "N roots"
names(env_dbrda)[names(env_dbrda) == "C.roots"] <- "C roots"
names(env_dbrda)[names(env_dbrda) == "crop.biomass"] <- "crop biomass"
names(env_dbrda)[names(env_dbrda) == "weed.cover"] <- "weed cover"
names(env_dbrda)[names(env_dbrda) == "plant.height"] <- "plant height"
names(env_dbrda)[names(env_dbrda) == "N.shoots"] <- "N shoots"
names(env_dbrda)[names(env_dbrda) == "C.shoots"] <- "C shoots"
names(env_dbrda)[names(env_dbrda) == "root.DW"] <- "root DW"

#sites (=my samples)
data.scores_dbrda = as.data.frame(scores(dbrda, display = 'sites')) #save to dataframe

#extract species
species.scores_dbrda = as.data.frame(scores(dbrda, display = 'species'))

#add grouping variable to dataframe
data.scores_dbrda <- cbind(as.data.frame(data.scores_dbrda), farming_system = env_dbrda$farming_system, time = env_dbrda$time)
head(data.scores_dbrda)

#rename columns 
names(data.scores_dbrda)[names(data.scores_dbrda) == "farming_system"] <- "Farming system"
names(data.scores_dbrda)[names(data.scores_dbrda) == "time"] <- "Sampling time"

set.seed(1722)
envfit_dbrda <- envfit(dbrda, env_dbrda, strata=veganSample_AMF_genus$block,na.rm = TRUE, display="lc")
envfit_dbrda

#extract scores from environment
env.scores_dbrda <- as.data.frame(scores(envfit_dbrda, display = "vectors"))
env.scores_dbrda <- cbind(env.scores_dbrda, env.variables = rownames(env.scores_dbrda), pval = envfit_dbrda$vectors$pvals)

head(env.scores_dbrda)
sig.env_dbrda <- subset(env.scores_dbrda, pval<=0.05) #extract only significant soil parameters
View(sig.env_dbrda)

#vectors as continuous, factors as categorical
#vectors are soil parameters and factors (centroids) are time + farming system
en_coord_cont_dbrda = as.data.frame(scores(envfit_dbrda, "vectors")) * ordiArrowMul(envfit_dbrda)
en_coord_cat_dbrda = as.data.frame(scores(envfit_dbrda, "factors")) * ordiArrowMul(envfit_dbrda)

#rename factors

rownames(en_coord_cat_dbrda)[rownames(en_coord_cat_dbrda) == "farming_systemBIODYN"] <- "BIODYN"
rownames(en_coord_cat_dbrda)[rownames(en_coord_cat_dbrda) == "farming_systemCONMIN"] <- "CONMIN"
rownames(en_coord_cat_dbrda)[rownames(en_coord_cat_dbrda) == "time4w"] <- "4 weeks"
rownames(en_coord_cat_dbrda)[rownames(en_coord_cat_dbrda) == "time13w"] <- "13 weeks"

#envfit for species
spp.fit <- envfit(dbrda, env_dbrda, strata=veganSample_AMF_genus$block,na.rm = TRUE, display="lc")

```

## plot dbRDA with environmental parameters 

```{r plot dbRDA and environmental parameters, echo=F}

dbrda_katja_env <- dbrda_katja + geom_segment(data = sig.env_dbrda,
               aes(x = 0, xend = dbRDA1,
                   y = 0, yend = dbRDA2),
               arrow = arrow(length = unit(2.5, "mm")),
               colour = "black",
               alpha=1,
               size=0.6,
               lineend ="round",
               linejoin="round") + 
               ggrepel::geom_text_repel(data = sig.env_dbrda, aes(x=dbRDA1, y=dbRDA2, label =env.variables), cex = 4,                color="black", direction = "both", segment.size = 0.25) + geom_point(data = en_coord_cat_dbrda, aes(x                = dbRDA1, y = dbRDA2), shape = "diamond", size = 8, alpha = 0.8, colour = "black") + geom_text(data =                en_coord_cat_dbrda, aes(x = dbRDA1, y = dbRDA2), label = row.names(en_coord_cat_dbrda), colour =                     "black", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor =                           element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

print(dbrda_katja_env)

```

## add species to dbrda

```{r add species to dbrda, echo=F}

df1  <- data.frame(smry$species[,1:2]) #dbrda1, 2
df2  <- data.frame(smry$centroids[,1:2]) #loadings for dbrda1, 2

#rename rows
rownames(df2)[rownames(df2) == "farming_systemBIODYN"] <- "BIODYN"
rownames(df2)[rownames(df2) == "farming_systemCONMIN"] <- "CONMIN"
rownames(df2)[rownames(df2) == "time4w"] <- "4 weeks"
rownames(df2)[rownames(df2) == "time13w"] <- "13 weeks"

rda.plot <- ggplot(df1, aes(x=dbRDA1, y=dbRDA2)) + 
  geom_label_repel(aes(label=rownames(df1)),
  point.padding = 0.5, box.padding   = 0.35, segment.color = 'grey50') + 
  geom_hline(yintercept=0, linetype="dotted") +
  geom_vline(xintercept=0, linetype="dotted") +
  coord_fixed() + theme_bw() + geom_segment(data=df1, aes(x=0, xend=dbRDA1, y=0, yend=dbRDA2),
  color="black", arrow=arrow(length=unit(0.01,"npc"))) + xlab("dbRDA1 (18.5%)") + ylab("dbRDA2 (7.7% )") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
  
#with this command we add arrows
rda.biplot <- rda.plot + geom_segment(data=df2, aes(x=0, xend=dbRDA1, y=0, yend=dbRDA2), 
              color="steelblue", arrow=arrow(length=unit(0.01,"npc"))) +
              geom_text_repel(data=df2,
              aes(x=dbRDA1,y=dbRDA2,label=rownames(df2),
              hjust=0.5*(1-sign(dbRDA1)),vjust=0.5*(1-sign(dbRDA2))), color="steelblue", size=4)


dbrda.biplot <- rda.plot + geom_text_repel(data=df2,
               aes(x=dbRDA1,y=dbRDA2,label=rownames(df2),
hjust=0.5*(1-sign(dbRDA1)),vjust=0.5*(1-sign(dbRDA2))),color="steelblue", size=4) + xlab("dbRDA1 (18.5%)") + ylab("dbRDA2 (7.7% )") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

dbrda.biplot

ggsave("results/dbrda_a.tiff", width = 20, height = 10, units = "cm")
ggsave("results/dbrda_a.jpeg", width = 20, height = 10, units = "cm")
ggsave("results/dbrda_a.pdf", width = 20, height = 10, units = "cm")

```

#dbrda dots 
```{r}

#Use min.segment.length = 0 to draw all line segments, no matter how short they are.
#Use min.segment.length = Inf to never draw any line segments, no matter how long they are.

#without arrows 
rda_points <- ggplot(df1, aes(x=dbRDA1, y=dbRDA2)) + 
    geom_label_repel(aes(label=rownames(df1)),
                     point.padding = 0.5, box.padding   = 0.35, segment.color = 'grey50', min.segment.length = 0) + geom_point(color="black", size = 2.5) +
    geom_hline(yintercept=0, linetype="dotted") +
    geom_vline(xintercept=0, linetype="dotted") +
    coord_fixed() + theme_bw() + xlab("dbRDA1 (18.5%)") + ylab("dbRDA2 (7.7% )") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

#with environmental variables 
rda_points + geom_segment(data = sig.env_dbrda,
                          aes(x = 0, xend = dbRDA1,
                              y = 0, yend = dbRDA2),
                          arrow = arrow(length = unit(2.5, "mm")),
                          colour = "grey",
                          alpha=1,
                          size=0.6,
                          lineend ="round",
                          linejoin="round") + 
    ggrepel::geom_text_repel(data = sig.env_dbrda, aes(x=dbRDA1, y=dbRDA2, label =env.variables), cex = 4.5,                color="grey", direction = "both", segment.size = 0.25)

```


## plot dbRDA with species

```{r plot dbrda and species, echo=F}

#plot diamonds
rda.plot +  geom_point(data = en_coord_cat_dbrda, aes(x = dbRDA1, y = dbRDA2), shape = "diamond", size = 8, alpha = 0.8, colour = "steelblue") + geom_text(data = en_coord_cat_dbrda, aes(x = dbRDA1, y = dbRDA2), label = row.names(en_coord_cat_dbrda), colour = "steelblue", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))

#plot environmental variables 
dbrda_env <- rda.plot + geom_segment(data = sig.env_dbrda,
               aes(x = 0, xend = dbRDA1,
                   y = 0, yend = dbRDA2),
               arrow = arrow(length = unit(2.5, "mm")),
               colour = "black",
               alpha=1,
               size=0.6,
               lineend ="round",
               linejoin="round") + 
               ggrepel::geom_text_repel(data = sig.env_dbrda, aes(x=dbRDA1, y=dbRDA2, label =env.variables), cex = 4,                color="steelblue", direction = "both", segment.size = 0.25)

ggsave("results/dbrda_b.tiff", width = 20, height = 10, units = "cm")
ggsave("results/dbrda_b.jpeg", width = 20, height = 10, units = "cm")
ggsave("results/dbrda_b.pdf", width = 20, height = 10, units = "cm")

#merge together
rda.plot +  geom_point(data = en_coord_cat_dbrda, aes(x = dbRDA1, y = dbRDA2), shape = "diamond", size = 8, alpha = 0.8, colour = "steelblue") + geom_text(data = en_coord_cat_dbrda, aes(x = dbRDA1, y = dbRDA2), label = row.names(en_coord_cat_dbrda), colour = "steelblue", fontface = "bold") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
geom_segment(data = sig.env_dbrda,
               aes(x = 0, xend = dbRDA1,
                   y = 0, yend = dbRDA2),
               arrow = arrow(length = unit(2.5, "mm")),
               colour = "black",
               alpha=1,
               size=0.6,
               lineend ="round",
               linejoin="round") + 
               ggrepel::geom_text_repel(data = sig.env_dbrda, aes(x=dbRDA1, y=dbRDA2, label =env.variables), cex = 4,                color="blue", direction = "both", segment.size = 0.25)
  
```

```{r}

library(cowplot)
dbrda.final <- plot_grid(dbrda.biplot, dbrda_env, labels = "AUTO")
save_plot("results/dbrda_final.tiff", dbrda.final, ncol = 2)
save_plot("results/dbrda_final.jpeg", dbrda.final, ncol = 2)
save_plot("results/dbrda_final.pdf", dbrda.final, ncol = 2)

```

# ALPHA diversity indices

```{r alpha_div, echo=F}

#the alpha diversity metrics were calculated based on samples rarefied to the lowest number of reads, calcualted per sample
#alpha diversity estimates on rarefied data
#we calculate Observed and Shannon for each sample

alpha_div <- data.frame(estimate_richness(physeq_unite_Filtered_ASV, split=TRUE, measures = c("Observed", "Shannon")), sample_data(physeq_unite_Filtered_ASV))

alpha_div

plot_richness(physeq_unite_Filtered_ASV, x="treatment", color="farming_system", measures=c("Observed","Shannon")) + stat_boxplot(geom = "errorbar") + geom_boxplot() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) + xlab("Treatment") + ggtitle("AMF alpha diversity") + ylab("Alpha metrics")

write.csv(alpha_div, "results/alpha_div.csv")

```
**Observed ASV richness and Shannon diversity index are calculated.**

## prepare data for BRMS model

```{r prepare data for BRMS model, echo=F}

#import data
head(alpha_div)

data_brms <- alpha_div
head(data_brms)
str(data_brms)

data_brms$farming_system <- as.factor(data_brms$farming_system)
data_brms$treatment <- as.factor(data_brms$treatment)
data_brms$time <- as.factor(data_brms$time)

```

+ negbinominal and gaussian BRMS model for observed richness 

```{r BRMS for observed richness, echo=F}

observed_brms <-brm(formula = (Observed)~time*treatment*farming_system+(1|block/plot),
            data = data_brms,
            family="negbinomial",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, stepsize=0.09, max_treedepth = 15))

pp_check(observed_brms)

# Save an object to a file
saveRDS(observed_brms, "data/rds/observed_brms.rds")

#reload it so you don't have to run it again
observed_brms <- readRDS("data/rds/observed_brms.rds")

#check model fit 
yfit_observed <- data_brms$Observed
yrep_observed <- posterior_predict(observed_brms, draws = length(yfit_observed))

launch_shinystan(observed_brms)

#explore results
summary(observed_brms)
bayes_R2(observed_brms)

#marginal effects
marginal_effects(observed_brms)

```
**BRMS model for Observed richness is created.**

## plot BRMS observed richness

```{r plot brms for observed richness, echo=F}

newdat <- expand.grid(
  time = factor(c("4w", "13w"),levels=levels(data_brms$time)), 
  farming_system = factor(c("BIODYN", "CONMIN"), levels=levels(data_brms$farming_system)),
  treatment = factor(c("C","R", "RC"),levels=levels(data_brms$treatment)))
head(newdat)

#use the function "fitted" from brms to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_observed = fitted(observed_brms, newdata = newdat, robust=T, re_formula = NA,summary = TRUE, probs = c(0.025, 0.975))  
fit_observed

# no colnames, I add them manually
fitdf_observed_plot = cbind(newdat, fit_observed)
fitdf_observed_plot

#change order of treatment in fitted values
fitdf_observed_plot$treatment<-factor(fitdf_observed_plot$treatment, levels=c("R", "RC", "C"))
data_brms$treatment<-factor(data_brms$treatment, levels=c("R", "RC", "C"))

#prepare and extract results table
fitdf_observed<-fitdf_observed_plot[, c("farming_system","treatment", "time", "Estimate","Q2.5","Q97.5")]
write.table(fitdf_observed,"results/observed_brms.txt",sep="\t",quote=F)

#this is the dataframe i will use for plotting
pos.dodge <- position_dodge(0.7) # move them .7 to the left and right

#make the final plot, without raw data
fitted_observed<-ggplot(data = fitdf_observed_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values=wes_palette(n=3, name="Moonrise2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  facet_grid(~time)+ 
  ylab("Observed ASV")+theme_bw()

print(fitted_observed)
ggsave("results/observed_brms.tiff", width = 18, height = 10, units = "cm")
ggsave("results/observed_brms.jpeg", width = 18, height = 10, units = "cm")
ggsave("results/observed_brms.pdf", width = 18, height = 10, units = "cm")

```


```{r, echo=F}
#get location of the means including credible intervals from here:
fitdf_observed_plot<-as.data.frame(fitdf_observed_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_observed = as.data.frame(fitted(observed_brms, newdata=newdat,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat$farming_system, newdat$treatment, newdat$time))

colnames(fit_observed)<-colnames
names(fit_observed)

#calculate the mean differences between systems across the roofs for 4w
BIODYN_brms_obs_T1<-(fit_observed$BIODYN.C.4w+fit_observed$BIODYN.RC.4w+fit_observed$BIODYN.R.4w)/3
mean(BIODYN_brms_obs_T1)
sd(BIODYN_brms_obs_T1)
CONMIN_brms_obs_T1<-(fit_observed$CONMIN.C.4w+fit_observed$CONMIN.RC.4w+fit_observed$CONMIN.R.4w)/3
mean(CONMIN_brms_obs_T1)
sd(CONMIN_brms_obs_T1)

diff_brms_obs_T1<-(BIODYN_brms_obs_T1-CONMIN_brms_obs_T1)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_obs_T1, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_obs_T1)

#calculate the mean differences between systems across the roofs for 13w
BIODYN_brms_obs_T3<-(fit_observed$BIODYN.C.13w+fit_observed$BIODYN.RC.13w+fit_observed$BIODYN.R.13w)/3
mean(BIODYN_brms_obs_T3)
sd(BIODYN_brms_obs_T3)
CONMIN_brms_obs_T3<-(fit_observed$CONMIN.C.13w+fit_observed$CONMIN.RC.13w+fit_observed$CONMIN.R.13w)/3
mean(CONMIN_brms_obs_T3)
sd(CONMIN_brms_obs_T3)

diff_brms_obs_T3<-(BIODYN_brms_obs_T3-CONMIN_brms_obs_T3)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_obs_T3, probs=c(0.025, 0.5, 0.975)),2)
#this is a clear result, it includes zero so its likely that there is no difference
mean(diff_brms_obs_T3)

```

## BRMS model for Shannon index 

```{r BRMS for Shannon, echo=F}

shannon_brms <-brm(formula = (Shannon)~time*treatment*farming_system+(1|block/plot),
            data = data_brms,
            family="gaussian",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, max_treedepth = 15))

pp_check(shannon_brms)

# Save an object to a file
saveRDS(shannon_brms, "data/rds/shannon_brms.rds")

#reload it so you don't have to run it again
shannon_brms <- readRDS("data/rds/shannon_brms.rds")

#check model fit 
yfit_shannon <- data_brms$Shannon
yrep_shannon <- posterior_predict(shannon_brms, draws = length(yfit_shannon))

launch_shinystan(shannon_brms)

#explore results
summary(shannon_brms)

#marginal effects
marginal_effects(shannon_brms)

```
**BRMS model for Shannon index is created.**

## plot brms for Shannon index

```{r plot brms for Shannon, echo=F}

#use the function "fitted" from brms
#to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_shannon = fitted(shannon_brms, newdata = newdat, robust=T, re_formula = NA,summary = TRUE,probs = c(0.025, 0.975))
fit_shannon

# no colnames, I add them manually
fitdf_shannon_plot = cbind(newdat, fit_shannon)
fitdf_shannon_plot

#change order of treatment in fitted values
fitdf_shannon_plot$treatment<-factor(fitdf_shannon_plot$treatment, levels=c("R", "RC", "C"))

#prepare and extract results table
fitdf_shannon<-fitdf_shannon_plot[, c("farming_system","treatment", "time", "Estimate","Q2.5","Q97.5")]
write.table(fitdf_shannon,"results/shannon_brms.txt",sep="\t",quote=F)

#make the final plot
fitted_shannon<-ggplot(data = fitdf_shannon_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values=wes_palette(n=3, name="Moonrise2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  facet_grid(~time) + 
  ylab("Shannon diversity index")+theme_bw()

print(fitted_shannon)
ggsave("results/shannon_brms.tiff", width = 18, height = 10, units = "cm")
ggsave("results/shannon_brms.jpeg", width = 18, height = 10, units = "cm")
ggsave("results/shannon_brms.pdf", width = 18, height = 10, units = "cm")

```

```{r, echo=F}

#get location of the means including credible intervals from here:
fitdf_shannon_plot<-as.data.frame(fitdf_shannon_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_shannon = as.data.frame(fitted(shannon_brms, newdata=newdat,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat$farming_system, newdat$treatment, newdat$time))

colnames(fit_shannon)<-colnames
names(fit_shannon)

#calculate the mean differences between systems across the roofs at 4w
BIODYN_brms_shannon_T1<-(fit_shannon$BIODYN.C.4w+fit_shannon$BIODYN.RC.4w+fit_shannon$BIODYN.R.4w)/3
mean(BIODYN_brms_shannon_T1)
sd(BIODYN_brms_shannon_T1)
CONMIN_brms_shannon_T1<-(fit_shannon$CONMIN.C.4w+fit_shannon$CONMIN.RC.4w+fit_shannon$CONMIN.R.4w)/3
mean(CONMIN_brms_shannon_T1)
sd(CONMIN_brms_shannon_T1)

diff_brms_shannon_T1 <- (BIODYN_brms_shannon_T1-CONMIN_brms_shannon_T1)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_shannon_T1, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_shannon_T1)


#calculate the mean differences between systems across the roofs at 13w
BIODYN_brms_shannon_T3<-(fit_shannon$BIODYN.C.13w+fit_shannon$BIODYN.RC.13w+fit_shannon$BIODYN.R.13w)/3
mean(BIODYN_brms_shannon_T3)
sd(BIODYN_brms_shannon_T3)
CONMIN_brms_shannon_T3<-(fit_shannon$CONMIN.C.13w+fit_shannon$CONMIN.RC.13w+fit_shannon$CONMIN.R.13w)/3
mean(CONMIN_brms_shannon_T3)
sd(CONMIN_brms_shannon_T3)

diff_brms_shannon_T3 <- (BIODYN_brms_shannon_T3-CONMIN_brms_shannon_T3)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_shannon_T3, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_shannon_T3)

```

# INDICATOR SPECIES ANALYSIS
+ to investigate if there are indicators of the farming system, the time of sampling and drought treatment

```{r indicator species analysis, echo=F}

#Transfer Phyloseq into a vegan ASV table
ind_analysis <- psotu2veg(physeq_AMF)

#meta file for indicators 
ind_meta <- as(sample_data(physeq_AMF), "data.frame")

ind_time <- ind_meta$time
ind_treatment <- ind_meta$treatment
ind_system <- ind_meta$farming_system

length(unique(ind_treatment))
length(unique(ind_time))
length(unique(ind_system))

#A = 1.0000: Species is a good indicator for the Group, because it occurs ONLY in sites, belonging to this group
#B: proportion of sites belonging to the group that contain the species.

#indicators for farming system 

set.seed(1809)
indval_System = multipatt(ind_analysis,ind_system,func = "IndVal", control = how(nperm=999))
summary(indval_System,indvalcomp=TRUE)

#indicators for drought treatment

set.seed(1809)
indval_Treatment = multipatt(ind_analysis,ind_treatment,func = "IndVal", control = how(nperm=999))
summary(indval_Treatment,indvalcomp=TRUE)

#indicators for the time of sampling

set.seed(1809)
indval_Time = multipatt(ind_analysis,ind_time,func = "IndVal", control = how(nperm=999))
summary(indval_Time,indvalcomp=TRUE)

```

```{r separate indicators for drought within farming system}

ind_analysis_biodyn <- ind_analysis[-c(1,2,3,7,8,15,16,17,21,22,23,24,25,26,30,31,32,38,39,40,44,45,46),]

ind_analysis_conmin <- ind_analysis[-c(4,5,6,9,10,11,12,13,14,18,19,20,27,28,29,33,34,35,36,37,41,42,43),]

#meta file for the indicators 
indicators_biodyn <- droplevels(subset(ind_meta, ind_meta$farming_system=="BIODYN"))
indicators_conmin <- droplevels(subset(ind_meta, ind_meta$farming_system=="CONMIN"))

biodyn_treatment <- indicators_biodyn$treatment
conmin_treatment <- indicators_conmin$treatment

biodyn_drought <- multipatt(ind_analysis_biodyn,biodyn_treatment,func = "IndVal", control = how(nperm=999))
summary(biodyn_drought,indvalcomp=TRUE)

conmin_drought <- multipatt(ind_analysis_conmin,conmin_treatment,func = "IndVal", control = how(nperm=999))
summary(conmin_drought,indvalcomp=TRUE) 

```

```{r separate indicators for drought within times}

ind_analysis_T1 <- ind_analysis[-c(24:46),]

ind_analysis_T3 <- ind_analysis[-c(1:23),]

#meta file for the indicators 
indicators_T1 <- droplevels(subset(ind_meta, ind_meta$time=="4w"))
indicators_T3 <- droplevels(subset(ind_meta, ind_meta$time=="13w"))

T1_treatment <- indicators_T1$treatment
T3_treatment <- indicators_T3$treatment

T1_drought <- multipatt(ind_analysis_T1,T1_treatment,func = "IndVal", control = how(nperm=999))
summary(T1_drought,indvalcomp=TRUE)

T3_drought <- multipatt(ind_analysis_T3,T3_treatment,func = "IndVal", control = how(nperm=999))
summary(T3_drought,indvalcomp=TRUE)

```
## indicators for the farming systems

```{r indicators system, echo=F}

indicators_system <- indval_System$sign

ind_BIODYN <- as.matrix(indicators_system[which(indicators_system$s.BIODYN == 1 & indicators_system$p.value < 0.05),])
ind_CONMIN <- as.matrix(indicators_system[which(indicators_system$s.CONMIN == 1 & indicators_system$p.value < 0.05),])

CONMIN_BIODYN <- rbind(ind_BIODYN, ind_CONMIN)
colnames(CONMIN_BIODYN)[1:2] <-c("BIODYN","CONMIN")

## Total number of indicator ASVs
length(unique(rownames(CONMIN_BIODYN)))

```

## indicators for the drought treatment

```{r indicators treatment, echo=F}

indicators_treatment <- indval_Treatment$sign

ind_R <- as.matrix(indicators_treatment[which(indicators_treatment$s.R == 1 & indicators_treatment$p.value < 0.05),])
ind_C <- as.matrix(indicators_treatment[which(indicators_treatment$s.C == 1 & indicators_treatment$p.value < 0.05),])
ind_RC <- as.matrix(indicators_treatment[which(indicators_treatment$s.RC == 1 & indicators_treatment$p.value < 0.05),])

R_C_RC <- rbind(ind_R, ind_C, ind_RC)
colnames(R_C_RC)[1:3] <-c("C","R","RC")

## Total number of indicator OTUS
length(unique(rownames(R_C_RC)))

```

## indicators for the time of sampling

```{r indicators time}

indicators_time <- indval_Time$sign

ind_T1 <- as.matrix(indicators_time[which(indicators_time$s.T1 == 1 & indicators_time$p.value < 0.05),])
ind_T3 <- as.matrix(indicators_time[which(indicators_time$s.T3 == 1 & indicators_time$p.value < 0.05),])

T1_T3 <- rbind(ind_T1, ind_T3)
colnames(T1_T3)[1:2] <-c("T1", "T3")

## Total number of indicator OTUS
length(unique(rownames(T1_T3)))

```

# TAXONOMIC COMPOSITION on different taxonomic levels
+ here we take a file with all ASVs with the relative abundances

## Class level
+ 3 classes 
+ Glomeromycetes as the most abundant 

```{r investigate class level, echo=F}

#taxonomy_AMF

unique(tax_table(unite_ASV_rel)[,3])

subset_class <- subset_taxa(unite_ASV_rel, Class %in% c("c__Paraglomeromycetes", "c__Glomeromycetes", "c__Archaeosporomycetes")) 

subset_class
subset_class = prune_samples(sample_sums(subset_class)>0, subset_class)

data_class <- subset_class %>%
              tax_glom(taxrank = "Class") %>%     #agglomerate at class level
              psmelt() %>%                        #melt to long format
              arrange(Class)                      #sort data frame alphabetically by class

# change the order of varible levels with factor()
data_class$treatment <- factor(data_class$treatment, levels = c("R", "RC", "C"))

#plot AMF class 
ggplot(data_class) + geom_col(mapping = aes(x = treatment:farming_system, y = Abundance, fill = Class), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Treatment:Farming system")  + facet_wrap(~time) + scale_fill_discrete(name="AMF class", breaks=c("c__Archaeosporomycetes", "c__Glomeromycetes", "c__Paraglomeromycetes"), labels=c("Archaeosporomycetes", "Glomeromycetes", "Paraglomeromycetes")) + ylab("Proportion")

```

## All orders 
+ there are 5 orders

```{r all orders AMF, echo=F}

unique(tax_table(unite_ASV_rel)[,4])

subset_orders_AMF <- subset_taxa(unite_ASV_rel, Order %in% c("o__Diversisporales","o__Glomerales","o__Gigasporales", "o__Paraglomerales", "o__Archaeosporales"))

orders_AMF = prune_samples(sample_sums(subset_orders_AMF)>0, subset_orders_AMF)

orders_AMF <- orders_AMF %>%
              tax_glom(taxrank = "Order") %>%      #agglomerate at order level
              psmelt() %>%                         #melt to long format
              arrange(Order)


# change the order of varible levels with factor()
orders_AMF$treatment <- factor(orders_AMF$treatment, levels = c("R", "RC", "C"))

ggplot(orders_AMF) + geom_col(mapping = aes(x = treatment:farming_system, y = Abundance, fill = Order), position = "fill", show.legend = TRUE)  + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Treatment:Farming system") + scale_fill_discrete(name="AMF order", breaks=c("o__Archaeosporales", "o__Diversisporales", "o__Glomerales", "o__Paraglomerales"), labels=c("Archaeosporales", "Diversisporales", "Glomerales", "Paraglomerales")) +ylab("Proportion")

```

## Focus on Glomeromycetes class
+ focus on Glomeromyctes class, as it is the most abundant one and three orders within it

```{r Glomeromycetes class, echo=F}

unique(tax_table(unite_ASV_rel)[,4])

subset_orders <- subset_taxa(unite_ASV_rel, Order %in% c("o__Diversisporales","o__Glomerales","o__Gigasporales"))

glomeromycetes_orders = prune_samples(sample_sums(subset_orders)>0, subset_orders)

glomeromycetes_orders <- glomeromycetes_orders %>%
                         tax_glom(taxrank = "Order") %>%      #agglomerate at order level
                         psmelt() %>%                         #melt to long format
                         arrange(Order)                       #sort data frame alphabetically by order

# change the order of varible levels with factor()
glomeromycetes_orders$treatment <- factor(glomeromycetes_orders$treatment, levels = c("R", "RC", "C"))

ggplot(glomeromycetes_orders) + geom_col(mapping = aes(x = treatment:farming_system, y = Abundance, fill = Order), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_fill_discrete(name="Glomeromycetes class", breaks=c("o__Diversisporales", "o__Gigasporales", "o__Glomerales"), labels=c("Diversisporales", "Gigasporales", "Glomerales")) + facet_wrap(~time) + ylab("Proportion")

```

## Family level 
+ there are 8 families 

```{r family level, echo=F}

unique(tax_table(unite_ASV_rel)[,5])

subset_family <- subset_taxa(unite_ASV_rel, Family %in% c("f__Archaeosporaceae", "f__Glomeraceae", "f__Diversisporaceae", "f__Ambisporaceae", "f__Acaulosporaceae", "f__Paraglomeraceae", "f__Gigasporaceae", "f__Claroideoglomeraceae")) 

subset_family
subset_family = prune_samples(sample_sums(subset_family)>0, subset_family)

data_family <- subset_family %>%
               tax_glom(taxrank = "Family") %>%
               psmelt() %>%
               arrange(Family)

# change the order of varible levels with factor()
data_family$treatment <- factor(data_family$treatment, levels = c("R", "RC", "C"))

ggplot(data_family) + geom_col(mapping = aes(x = treatment:farming_system, y = Abundance, fill = Family), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Treatment:Farming system") + scale_fill_discrete(name="AMF family", breaks=c("f__Acaulosporaceae", "f__Ambisporaceae", "f__Archaeosporaceae", "f__Claroideoglomeraceae", "f__Diversisporaceae", "f__Gigasporaceae",
"f__Glomeraceae",  "f__Paraglomeraceae"), labels=c("Acaulosporaceae", "Ambisporaceae", "Archaeosporaceae", "Claroideoglomeraceae", "Diversisporaceae", "Gigasporaceae", "Glomeraceae", "Paraglomeraceae")) + facet_wrap(~time) + ylab("Proportion")

```

## Genus level 
+ 12 genera
+ here we include NA

```{r genus level, echo=F}

#here we will include unassigned

unique(tax_table(unite_ASV_rel)[,6])

subset_genera <- subset_taxa(unite_ASV_rel, Genus %in% c("g__Acaulospora", "g__Glomus", "g__Palaeospora", "NA", "g__Diversispora", "g__Funneliformis", "g__Paraglomus", "g__Ambispora", "g__Claroideoglomus", "g__Gigaspora", "g__Dominikia", "g__Archaeospora", "g__Septoglomus"))

subset_genera = prune_samples(sample_sums(subset_genera)>0, subset_genera)

data_genera <- subset_genera %>%
               tax_glom(taxrank = "Genus") %>%
               psmelt() %>%
               arrange (Genus)

#include NA
data_genera_NA <- subset_genera %>%
                  psmelt() %>%
                  arrange (Genus)

# change the order of varible levels with factor()
data_genera$treatment <- factor(data_genera$treatment, levels = c("R", "RC", "C"))
data_genera$time <- factor(data_genera$time, levels=c("4w", "13w"), labels = c("4 weeks", "13 weeks"))
data_genera_NA$treatment <- factor(data_genera_NA$treatment, levels = c("R", "RC", "C"))

#treatment + farming system 
ggplot(data_genera) + geom_col(mapping = aes(x = treatment:farming_system, y = Abundance, fill = Genus), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Treatment:Farming system") + scale_fill_discrete(name="AMF genera", breaks=c("g__Acaulospora", "g__Ambispora", "g__Archaeospora", "g__Claroideoglomus", "g__Diversispora", "g__Dominikia", "g__Funneliformis", "g__Gigaspora", "g__Glomus", "g__Palaeospora", "g__Paraglomus", "g__Septoglomus"), labels=c("Acaulospora", "Ambispora", "Archaeospora", "Claroideoglomus", "Diversispora", "Dominikia", "Funneliformis", "Gigaspora", "Glomus", "Palaeospora", "Paraglomus", "Septoglomus")) + facet_wrap(~time) + ylab("Proportion")

#treatment 
ggplot(data_genera) + geom_col(mapping = aes(x = treatment, y = Abundance, fill = Genus), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Treatment") + scale_fill_discrete(name="AMF genera", breaks=c("g__Acaulospora", "g__Ambispora", "g__Archaeospora", "g__Claroideoglomus", "g__Diversispora", "g__Dominikia", "g__Funneliformis", "g__Gigaspora", "g__Glomus", "g__Palaeospora", "g__Paraglomus", "g__Septoglomus"), labels=c("Acaulospora", "Ambispora", "Archaeospora", "Claroideoglomus", "Diversispora", "Dominikia", "Funneliformis", "Gigaspora", "Glomus", "Palaeospora", "Paraglomus", "Septoglomus")) + facet_wrap(~time) + ylab("Proportion")

#farming system
ggplot(data_genera) + geom_col(mapping = aes(x = farming_system, y = Abundance, fill = Genus), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 35, hjust = 1)) + scale_x_discrete(name ="Farming system") + scale_fill_discrete(name="AMF genera", breaks=c("g__Acaulospora", "g__Ambispora", "g__Archaeospora", "g__Claroideoglomus", "g__Diversispora", "g__Dominikia", "g__Funneliformis", "g__Gigaspora", "g__Glomus", "g__Palaeospora", "g__Paraglomus", "g__Septoglomus"), labels=c("Acaulospora", "Ambispora", "Archaeospora", "Claroideoglomus", "Diversispora", "Dominikia", "Funneliformis", "Gigaspora", "Glomus", "Palaeospora", "Paraglomus", "Septoglomus")) + facet_wrap(~time) + ylab("Proportion")

#with NA as others

plot_genus <- ggplot(data_genera_NA) + geom_col(mapping = aes(x = farming_system:treatment, y = Abundance, fill = Genus), position = "fill", show.legend = TRUE) + theme(axis.text.x = element_text(angle = 30, hjust = 1)) + scale_x_discrete(name ="Farming system:Drought treatment") + scale_fill_discrete(name="AMF genera", breaks=c("g__Acaulospora", "g__Ambispora", "g__Archaeospora", "g__Claroideoglomus", "g__Diversispora", "g__Dominikia", "g__Funneliformis", "g__Gigaspora", "g__Glomus", "g__Palaeospora", "g__Paraglomus", "g__Septoglomus", "<NA>"), labels=c("Acaulospora", "Ambispora", "Archaeospora", "Claroideoglomus", "Diversispora", "Dominikia", "Funneliformis", "Gigaspora", "Glomus", "Palaeospora", "Paraglomus", "Septoglomus", "Others")) + facet_wrap(~time) + ylab("Proportion of relative abundance") + scale_fill_manual(values = mycolors)


#facet_wrap(farming_system~time)
#facet_grid(time~farming_system)

genusAMF_abundance <- ggplot(data_genera) + geom_col(mapping = aes(x = treatment, y = Abundance, fill = Genus), position = "fill", show.legend = TRUE) + scale_x_discrete(name ="Drought treatment") + scale_fill_manual(values = c("#EAD3BF", "#AA9486", "#B6854D", "#7294D4","#F1BB7B", "#FD6467", "#9C964A", "#CDC08C", "#FAD77B","#F3DF6C", "#CEAB07","#D3DDDC"), name="AMF genera", breaks=c("g__Acaulospora", "g__Ambispora", "g__Archaeospora", "g__Claroideoglomus", "g__Diversispora", "g__Dominikia", "g__Funneliformis", "g__Gigaspora", "g__Glomus", "g__Palaeospora", "g__Paraglomus", "g__Septoglomus"), labels=c("Acaulospora", "Ambispora", "Archaeospora", "Claroideoglomus", "Diversispora", "Dominikia", "Funneliformis", "Gigaspora", "Glomus", "Palaeospora", "Paraglomus", "Septoglomus")) + ylab("Proportion of relative abundance") + theme_bw() + facet_wrap(time~farming_system)

print(genusAMF_abundance)

ggsave("results/stacked_bar_genus.tiff", width = 15, height = 10, units = "cm")
ggsave("results/stacked_bar_genus.jpeg", width = 15, height = 10, units = "cm")
ggsave("results/stacked_bar_genus.pdf", width = 15, height = 10, units = "cm")

```

# add figure for water

```{r water content}

View(water_content)

data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}

water_summary <- data_summary(water_content, varname="water", 
                 groupnames=c("treatment", "farming_system", "time"))

water_summary$farming_system <- as.factor(water_summary$farming_system)

water_summary$time <- as.factor(water_summary$time)
#plot 4w before 123w
water_summary$time_order <- factor(water_summary$time, levels=c("4w", "13w"), labels = c("4 weeks", "13 weeks"))

water_summary$treatment<-factor(water_summary$treatment, levels=c("R", "RC", "C"))

p_water <- ggplot(water_summary, aes(farming_system, water)) +
    geom_errorbar(
    aes(ymin = water-sd, ymax = water+sd, color = treatment),
    position = position_dodge(0.3), width = 0.2
    ) +
    geom_point(aes(color = treatment), position = position_dodge(0.3), size=2) +     facet_grid(.~time_order) + theme_bw() + xlab("Farming system") + ylab("WC / WHC") + scale_color_manual(values = c("#B6854D", "#8D8680", "#0F0D0E")) + labs(color = "Drought treatment")

p_water
ggsave("results/water.tiff", width = 20, height = 10, units = "cm", dpi=500)
ggsave("results/water.jpeg", width = 20, height = 10, units = "cm", dpi=500)
ggsave("results/water.pdf", width = 20, height = 10, units = "cm", dpi=500)

```

# BRMS model for water

```{r brms model for water}

brms_water <- brm(formula = (water)~time*treatment*farming_system+(1|block/plot),
            data = water_content,
            family="gaussian",
            seed = 1041,
            warmup = 2000,#The number of warmup iterations should 
            #not be larger than iter and the default is iter/2.
            iter = 6000, 
            chains = 5,
            control = list(adapt_delta = 0.999, max_treedepth = 15))

pp_check(brms_water)

# Save an object to a file
saveRDS(brms_water, "data/rds/brms_water.rds")

#reload it so you don't have to run it again
brms_water <- readRDS("data/rds/brms_water.rds")

#check model fit 
yfit_water <- water_content$water
yrep_water<- posterior_predict(brms_water, draws = length(yfit_water))

launch_shinystan(brms_water)

#explore results
summary(brms_water)

#marginal effects
marginal_effects(brms_water)

```

## plot BRMS water

```{r plot brms for water, echo=F}

newdat_water <- expand.grid(
  time = factor(c("4w", "13w"),levels=levels(water_summary$time)), 
  farming_system = factor(c("BIODYN", "CONMIN"), levels=levels(water_summary$farming_system)),
  treatment = factor(c("C","R","RC"),levels=levels(water_summary$treatment)))
head(newdat_water)

#use the function "fitted" from brms to get fitted means of the posterior for treatments
#It summarizes from the posterior and calculates meadian with 95% credible interval
#= we are 95% sure that the real estimate is within this interval

fit_water = fitted(brms_water, newdata = newdat_water, robust=T, re_formula = NA,summary = TRUE, probs = c(0.025, 0.975))  
fit_water

# no colnames, I add them manually
fitdf_water_plot = cbind(newdat_water, fit_water)
fitdf_water_plot

#change order of treatment in fitted values
fitdf_water_plot$treatment<-factor(fitdf_water_plot$treatment, levels=c("R", "RC", "C"))
water_summary$treatment<-factor(water_summary$treatment, levels=c("R", "RC", "C"))

#prepare and extract results table
fitdf_water<-fitdf_water_plot[, c("farming_system","treatment", "time", "Estimate","Q2.5","Q97.5")]
write.table(fitdf_water,"results/water_brms.txt",sep="\t",quote=F)

#this is the dataframe i will use for plotting
pos.dodge <- position_dodge(0.7) # move them .7 to the left and right

#make the final plot, without raw data
fitted_water<-ggplot(data = fitdf_water_plot, 
                     aes(x = farming_system,
                         y =Estimate,
                         color = treatment,
                         shape=treatment)) +
  geom_point(size=6,position=pos.dodge) +
  geom_errorbar(aes(ymin=Q2.5,
                    ymax=Q97.5),
                width=0,
                position=pos.dodge,
                size=1.2, show.legend = F) +
  scale_color_manual(values=wes_palette(n=3, name="Moonrise2"), name  ="") +
  scale_shape_manual(values = c(15, 17, 19),name  ="")+
  xlab("Farming system")+
  facet_grid(~time)+ 
  ylab("Observed ASV")+theme_bw()

```

## BRMS summary water

```{r, echo=F}
#get location of the means including credible intervals from here:
fitdf_water_plot<-as.data.frame(fitdf_water_plot)

#if you want, you can also calculate differences between treatments based on the posterior
##calucate differences with CrI, you need the whole posterior
#the whole posterior distribution for me. I save it as dataframe (fit1)

fit_water = as.data.frame(fitted(brms_water, newdata=newdat_water,
                           re_formula = NA, summary = FALSE))

#this is the whole posterior, not summarized as before

#add names
colnames<-as.character(interaction(newdat_water$farming_system, newdat_water$treatment, newdat_water$time))

colnames(fit_water)<-colnames
names(fit_water)

#calculate the mean differences between systems across the roofs for 4w
BIODYN_brms_water_T1<-(fit_water$BIODYN.C.4w+fit_water$BIODYN.RC.4w+fit_water$BIODYN.R.4w)/3
mean(BIODYN_brms_water_T1)
sd(BIODYN_brms_obs_T1)
CONMIN_brms_water_T1<-(fit_water$CONMIN.C.4w+fit_water$CONMIN.RC.4w+fit_water$CONMIN.R.4w)/3
mean(CONMIN_brms_water_T1)
sd(CONMIN_brms_water_T1)

diff_brms_water_T1<-(BIODYN_brms_water_T1-CONMIN_brms_water_T1)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_water_T1, probs=c(0.025, 0.5, 0.975)),2)
#this is a result
mean(diff_brms_water_T1)

#calculate the mean differences between systems across the roofs for 13w
BIODYN_brms_water_T3<-(fit_water$BIODYN.C.13w+fit_water$BIODYN.RC.13w+fit_water$BIODYN.R.13w)/3
mean(BIODYN_brms_water_T3)
sd(BIODYN_brms_water_T3)
CONMIN_brms_water_T3<-(fit_water$CONMIN.C.13w+fit_water$CONMIN.RC.13w+fit_water$CONMIN.R.13w)/3
mean(CONMIN_brms_water_T3)
sd(CONMIN_brms_water_T3)

diff_brms_water_T3<-(BIODYN_brms_water_T3-CONMIN_brms_water_T3)

#use the quantile function to get mean of that differences with CrI
round(quantile(diff_brms_water_T3, probs=c(0.025, 0.5, 0.975)),2)
#this is a clear result, it includes zero so its likely that there is no difference
mean(diff_brms_water_T3)

```

# Summary 
+ UNITE database was used for taxonomic assignments
+ samples with less than 500 were removed
+ TSS normalization
+ AMF diversity: neither farming system nor drought treatment did have an effect on the two diversity metrics, observed number of ASVs and Shannon’s index of AMF
+ AMF community composition: data agglomerated on genus level, farming effect over time 
+ farming and time effect need to be highlighted with constrained method 
+ except root.DW and PO4_P all soil parameters had significant relations on the AMF community composition
+ there are indicators for system (CONMIN), time of the sampling and drought treatment 

# Session Info

```{r sessionInfo, include=TRUE, echo=TRUE}
sessionInfo()
```

